<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">@import url(https://themes.googleusercontent.com/fonts/css?kit=tKLL3UDOEolLNLmIs0YRYA);ol{margin:0;padding:0}table td,table th{padding:0}.c142{-webkit-text-decoration-skip:none;color:#231f20;font-weight:400;text-decoration:line-through;vertical-align:super;text-decoration-skip-ink:none;font-size:8.3pt;font-family:"Arial";font-style:normal}.c92{margin-left:12.1pt;padding-top:18.1pt;text-indent:2.1pt;padding-bottom:0pt;line-height:1.015763521194458;text-align:left;margin-right:11.2pt}.c149{margin-left:11.7pt;padding-top:0.2pt;text-indent:0pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:0.3pt}.c194{margin-left:6.2pt;padding-top:5.5pt;text-indent:0.3pt;padding-bottom:0pt;line-height:0.8754744529724121;text-align:left;margin-right:2.6pt}.c51{color:#231f20;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:7pt;font-family:"Arial";font-style:normal}.c294{margin-left:20.7pt;padding-top:0.2pt;text-indent:-20.7pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:justify;margin-right:8.2pt}.c95{margin-left:25.9pt;padding-top:0.5pt;text-indent:-0.2pt;padding-bottom:0pt;line-height:1.0137923955917358;text-align:left;margin-right:25.1pt}.c190{margin-left:20.7pt;padding-top:0.2pt;text-indent:-20.7pt;padding-bottom:0pt;line-height:0.9220926761627197;text-align:left;margin-right:8pt}.c13{color:#0000ff;font-weight:400;text-decoration:none;vertical-align:super;font-size:10pt;font-family:"Arial";font-style:normal}.c52{margin-left:6.1pt;padding-top:11.4pt;text-indent:0.1pt;padding-bottom:0pt;line-height:1.0161088705062866;text-align:justify;margin-right:2pt}.c223{margin-left:6.1pt;padding-top:0.3pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.014415979385376;text-align:justify;margin-right:2pt}.c103{margin-left:28.1pt;padding-top:0pt;text-indent:-16.4pt;padding-bottom:0pt;line-height:0.9220961332321167;text-align:justify;margin-right:0.6pt}.c32{margin-left:0.2pt;padding-top:17.1pt;text-indent:-0.1pt;padding-bottom:0pt;line-height:0.8669092059135437;text-align:justify;margin-right:39.7pt}.c265{margin-left:7.1pt;padding-top:0.6pt;text-indent:10pt;padding-bottom:0pt;line-height:0.9386309385299683;text-align:justify;margin-right:0pt}.c8{color:#0000ff;font-weight:400;text-decoration:none;vertical-align:super;font-size:8.3pt;font-family:"Arial";font-style:normal}.c165{margin-left:6.2pt;padding-top:0.3pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.0040316581726074;text-align:justify;margin-right:2pt}.c229{margin-left:6.6pt;padding-top:0pt;text-indent:-0.4pt;padding-bottom:0pt;line-height:1.015305519104004;text-align:left;margin-right:2.3pt}.c2{color:#231f20;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:8pt;font-family:"Arial";font-style:normal}.c205{margin-left:6.2pt;padding-top:0.3pt;text-indent:9.9pt;padding-bottom:0pt;line-height:1.0158751010894775;text-align:justify;margin-right:1.9pt}.c286{margin-left:25.8pt;padding-top:0.2pt;text-indent:-16.7pt;padding-bottom:0pt;line-height:0.9221660494804382;text-align:left;margin-right:1.1pt}.c170{margin-left:28.1pt;padding-top:0.2pt;text-indent:-21pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:justify;margin-right:0.1pt}.c178{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:13.9pt;font-family:"Arial";font-style:italic}.c67{margin-left:16.7pt;padding-top:0.2pt;text-indent:-16.7pt;padding-bottom:0pt;line-height:0.9982525110244751;text-align:justify;margin-right:10pt}.c132{margin-left:6.2pt;padding-top:0.4pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.0157091617584229;text-align:justify;margin-right:1.9pt}.c6{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:5pt;font-family:"Times";font-style:normal}.c98{margin-left:16.4pt;padding-top:0.2pt;text-indent:-16.4pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:10.2pt}.c288{margin-left:25.5pt;padding-top:0pt;text-indent:-16.4pt;padding-bottom:0pt;line-height:0.9780585169792175;text-align:justify;margin-right:0.8pt}.c104{margin-left:266.2pt;padding-top:0pt;text-indent:-262.4pt;padding-bottom:0pt;line-height:0.9696372747421265;text-align:left;margin-right:144.5pt}.c129{margin-left:6.2pt;padding-top:0pt;text-indent:-0.1pt;padding-bottom:0pt;line-height:1.0144165754318237;text-align:justify;margin-right:1.9pt}.c99{margin-left:6.1pt;padding-top:11.3pt;text-indent:0.1pt;padding-bottom:0pt;line-height:1.014999508857727;text-align:justify;margin-right:0.8pt}.c10{color:#0000ff;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Arial";font-style:normal}.c183{margin-left:6.1pt;padding-top:11.3pt;text-indent:0.1pt;padding-bottom:0pt;line-height:0.9973804354667664;text-align:justify;margin-right:1.9pt}.c5{color:#231f20;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:7pt;font-family:"Arial";font-style:normal}.c0{color:#231f20;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:8pt;font-family:"Arial";font-style:normal}.c175{margin-left:0.1pt;padding-top:0pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.0149778127670288;text-align:justify;margin-right:8.1pt}.c4{color:#000000;font-weight:400;text-decoration:none;vertical-align:super;font-size:8.4pt;font-family:"Times";font-style:normal}.c85{margin-left:25.6pt;padding-top:0.2pt;text-indent:-16.5pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:1.3pt}.c19{color:#231f20;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:5pt;font-family:"Arial";font-style:normal}.c18{color:#0000ff;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Arial";font-style:normal}.c25{margin-left:12.8pt;padding-top:0.3pt;text-indent:0.1pt;padding-bottom:0pt;line-height:1.0137895345687866;text-align:left;margin-right:26.8pt}.c37{margin-left:5.1pt;padding-top:0.2pt;text-indent:0pt;padding-bottom:0pt;line-height:0.9220959544181824;text-align:left;margin-right:1.1pt}.c42{margin-left:17pt;padding-top:0.2pt;text-indent:-17pt;padding-bottom:0pt;line-height:0.9221717119216919;text-align:left;margin-right:10.2pt}.c57{margin-left:21.5pt;padding-top:0.2pt;text-indent:-13pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:7.4pt}.c214{margin-left:28.1pt;padding-top:0.2pt;text-indent:-21pt;padding-bottom:0pt;line-height:0.9221660494804382;text-align:left;margin-right:1.1pt}.c111{margin-left:25.6pt;padding-top:0pt;text-indent:-16.5pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:justify;margin-right:0.6pt}.c212{margin-left:6.1pt;padding-top:33.7pt;text-indent:0.3pt;padding-bottom:0pt;line-height:1.0146228075027466;text-align:left;margin-right:22.5pt}.c200{margin-left:17pt;padding-top:0.2pt;text-indent:-17pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:10.2pt}.c69{margin-left:25.9pt;padding-top:0.2pt;text-indent:-16.8pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:1.1pt}.c11{margin-left:17.2pt;padding-top:0.2pt;text-indent:-17.2pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:10.2pt}.c173{margin-left:6.3pt;padding-top:0pt;text-indent:0.1pt;padding-bottom:0pt;line-height:1.0142089128494263;text-align:justify;margin-right:0.8pt}.c60{margin-left:0.1pt;padding-top:11.4pt;text-indent:0.5pt;padding-bottom:0pt;line-height:1.0150114297866821;text-align:justify;margin-right:8pt}.c43{margin-left:0.1pt;padding-top:26.6pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.015724778175354;text-align:justify;margin-right:6.9pt}.c27{margin-left:11.7pt;padding-top:0.2pt;text-indent:0pt;padding-bottom:0pt;line-height:0.9220829606056213;text-align:left;margin-right:0.1pt}.c101{margin-left:0pt;padding-top:27.4pt;text-indent:0.3pt;padding-bottom:0pt;line-height:1.0146256685256958;text-align:left;margin-right:32.9pt}.c131{margin-left:6.9pt;padding-top:0pt;text-indent:-0.5pt;padding-bottom:0pt;line-height:1.0146228075027466;text-align:left;margin-right:33.9pt}.c47{margin-left:0.8pt;padding-top:0pt;text-indent:-0.5pt;padding-bottom:0pt;line-height:0.8363329768180847;text-align:left;margin-right:33.4pt}.c159{margin-left:3.5pt;padding-top:9pt;text-indent:-0.2pt;padding-bottom:0pt;line-height:1.0486053228378296;text-align:left;margin-right:65pt}.c31{color:#0000ff;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:8pt;font-family:"Arial";font-style:normal}.c232{margin-left:0.1pt;padding-top:9pt;text-indent:0.3pt;padding-bottom:0pt;line-height:1.0148289203643799;text-align:left;margin-right:50.8pt}.c68{color:#231f20;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:6pt;font-family:"Arial";font-style:normal}.c118{margin-left:21.1pt;padding-top:31.3pt;text-indent:-12.6pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:7.4pt}.c187{margin-left:29.1pt;padding-top:0.2pt;text-indent:-17.5pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:0.6pt}.c291{margin-left:0.1pt;padding-top:0.4pt;text-indent:9.8pt;padding-bottom:0pt;line-height:0.982357382774353;text-align:justify;margin-right:8pt}.c192{margin-left:0.4pt;padding-top:12.9pt;text-indent:-0.4pt;padding-bottom:0pt;line-height:0.8363358378410339;text-align:left;margin-right:18.1pt}.c137{margin-left:16.5pt;padding-top:0.2pt;text-indent:-16.5pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:9.7pt}.c273{margin-left:28.2pt;padding-top:0.2pt;text-indent:-16.5pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:0pt}.c199{margin-left:26.1pt;padding-top:0.2pt;text-indent:-21pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:1.1pt}.c202{margin-left:17.8pt;padding-top:0.2pt;text-indent:-17.8pt;padding-bottom:0pt;line-height:0.9220926761627197;text-align:left;margin-right:10.2pt}.c124{margin-left:17.2pt;padding-top:0.2pt;text-indent:-17.2pt;padding-bottom:0pt;line-height:0.9220926761627197;text-align:left;margin-right:10.2pt}.c55{margin-left:7.1pt;padding-top:11.4pt;text-indent:-0pt;padding-bottom:0pt;line-height:1.0143866539001465;text-align:justify;margin-right:0.1pt}.c109{margin-left:25.7pt;padding-top:0.3pt;text-indent:-13.2pt;padding-bottom:0pt;line-height:1.0143468379974365;text-align:justify;margin-right:16.8pt}.c30{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Arial";font-style:normal}.c234{margin-left:11.7pt;padding-top:0.2pt;text-indent:-0pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:42.2pt}.c16{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:11.5pt;font-family:"Arial";font-style:normal}.c250{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:normal}.c222{margin-left:7.1pt;padding-top:0pt;text-indent:0.2pt;padding-bottom:0pt;line-height:0.9612308740615845;text-align:justify;margin-right:0.1pt}.c284{margin-left:25.9pt;padding-top:0pt;text-indent:-16.8pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:justify;margin-right:1pt}.c230{margin-left:16.7pt;padding-top:0.2pt;text-indent:-16.7pt;padding-bottom:0pt;line-height:0.9221321940422058;text-align:justify;margin-right:10.2pt}.c119{margin-left:12.6pt;padding-top:0.5pt;text-indent:-6.5pt;padding-bottom:0pt;line-height:1.0146228075027466;text-align:left;margin-right:44pt}.c128{margin-left:25.7pt;padding-top:0.4pt;text-indent:-13.4pt;padding-bottom:0pt;line-height:1.0145728588104248;text-align:left;margin-right:9.4pt}.c9{margin-left:19.5pt;padding-top:0.2pt;text-indent:-11pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:7.4pt}.c115{margin-left:6.1pt;padding-top:0.4pt;text-indent:10.1pt;padding-bottom:0pt;line-height:0.9970251321792603;text-align:justify;margin-right:0.8pt}.c161{margin-left:28.2pt;padding-top:0.9pt;text-indent:-12.5pt;padding-bottom:0pt;line-height:0.9221349954605103;text-align:justify;margin-right:0.5pt}.c163{margin-left:20.5pt;padding-top:0pt;text-indent:-20.5pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:7.7pt}.c242{margin-left:28.7pt;padding-top:0.2pt;text-indent:-17pt;padding-bottom:0pt;line-height:0.9216901063919067;text-align:justify;margin-right:0.5pt}.c102{margin-left:12.6pt;padding-top:0.5pt;text-indent:-6.5pt;padding-bottom:0pt;line-height:1.0146256685256958;text-align:left;margin-right:44pt}.c150{margin-left:21pt;padding-top:0.2pt;text-indent:-12.4pt;padding-bottom:0pt;line-height:0.9220965504646301;text-align:justify;margin-right:7.6pt}.c215{margin-left:16.4pt;padding-top:0.2pt;text-indent:-16.4pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:justify;margin-right:10.2pt}.c20{margin-left:12.9pt;padding-top:0.5pt;text-indent:0pt;padding-bottom:0pt;line-height:1.0146256685256958;text-align:left;margin-right:16.7pt}.c256{margin-left:16.5pt;padding-top:0.2pt;text-indent:-16.5pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:justify;margin-right:10.2pt}.c87{margin-left:16.8pt;padding-top:0.2pt;text-indent:-16.8pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:justify;margin-right:10.2pt}.c63{margin-left:6.2pt;padding-top:11.4pt;text-indent:-0pt;padding-bottom:0pt;line-height:1.015609860420227;text-align:justify;margin-right:2pt}.c181{margin-left:28.7pt;padding-top:0.2pt;text-indent:-17pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:0.8pt}.c62{margin-left:0.1pt;padding-top:0pt;text-indent:0.2pt;padding-bottom:0pt;line-height:0.9740413427352905;text-align:justify;margin-right:8pt}.c282{margin-left:25.6pt;padding-top:0pt;text-indent:-16.5pt;padding-bottom:0pt;line-height:0.9230247139930725;text-align:left;margin-right:0.6pt}.c171{margin-left:25.4pt;padding-top:0.4pt;text-indent:-12.9pt;padding-bottom:0pt;line-height:1.0149635076522827;text-align:left;margin-right:14.7pt}.c114{margin-left:11.7pt;padding-top:0.2pt;text-indent:-0pt;padding-bottom:0pt;line-height:0.9220964312553406;text-align:left;margin-right:0.5pt}.c246{margin-left:28.6pt;padding-top:0.2pt;text-indent:-12.9pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:justify;margin-right:0.6pt}.c15{color:#231f20;font-weight:400;text-decoration:none;vertical-align:super;font-size:8.3pt;font-family:"Arial";font-style:normal}.c208{margin-left:6.2pt;padding-top:5.5pt;text-indent:-0.2pt;padding-bottom:0pt;line-height:0.8966454863548279;text-align:justify;margin-right:1.9pt}.c141{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:19.9pt;font-family:"Arial";font-style:normal}.c12{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Arial";font-style:normal}.c64{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:9.7pt;font-family:"Arial";font-style:normal}.c263{margin-left:28.9pt;padding-top:0.2pt;text-indent:-17.2pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:0.5pt}.c278{margin-left:25.4pt;padding-top:0.3pt;text-indent:-12.9pt;padding-bottom:0pt;line-height:1.0142089128494263;text-align:left;margin-right:10.3pt}.c216{margin-left:27.5pt;padding-top:0pt;text-indent:-20.4pt;padding-bottom:0pt;line-height:0.9221349954605103;text-align:justify;margin-right:0.6pt}.c89{margin-left:6.2pt;padding-top:0pt;text-indent:0.4pt;padding-bottom:0pt;line-height:1.0075929164886475;text-align:justify;margin-right:2pt}.c220{margin-left:27.8pt;padding-top:0.2pt;text-indent:-20.7pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:0.9pt}.c121{margin-left:202.9pt;padding-top:12.9pt;text-indent:-202.9pt;padding-bottom:0pt;line-height:0.8363202214241028;text-align:left;margin-right:0.4pt}.c75{color:#000000;font-weight:400;text-decoration:none;vertical-align:super;font-size:13.9pt;font-family:"Times";font-style:normal}.c218{margin-left:0.1pt;padding-top:0.4pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.0150246620178223;text-align:justify;margin-right:8.2pt}.c3{color:#231f20;font-weight:400;text-decoration:none;vertical-align:super;font-size:10pt;font-family:"Arial";font-style:normal}.c33{margin-left:6.2pt;padding-top:0pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.0153642892837524;text-align:justify;margin-right:2pt}.c255{margin-left:6.3pt;padding-top:11.5pt;text-indent:0.4pt;padding-bottom:0pt;line-height:1.0158371925354004;text-align:justify;margin-right:2.1pt}.c198{margin-left:7pt;padding-top:0pt;text-indent:0.2pt;padding-bottom:0pt;line-height:1.0148684978485107;text-align:justify;margin-right:0.1pt}.c117{margin-left:26.3pt;padding-top:0.2pt;text-indent:-21.2pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:1pt}.c123{margin-left:6.2pt;padding-top:0.3pt;text-indent:10pt;padding-bottom:0pt;line-height:1.0152747631072998;text-align:justify;margin-right:0.8pt}.c143{margin-left:26.3pt;padding-top:0.2pt;text-indent:-17.2pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:1.1pt}.c17{color:#231f20;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:7pt;font-family:"Arial";font-style:italic}.c94{margin-left:25.5pt;padding-top:0.2pt;text-indent:-16.4pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:1pt}.c249{margin-left:6.1pt;padding-top:0.3pt;text-indent:9.5pt;padding-bottom:0pt;line-height:1.0153636932373047;text-align:justify;margin-right:0.8pt}.c264{margin-left:28.2pt;padding-top:0.2pt;text-indent:-12.5pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:0.8pt}.c245{margin-left:6.2pt;padding-top:11.4pt;text-indent:-0pt;padding-bottom:0pt;line-height:1.014625072479248;text-align:justify;margin-right:2pt}.c1{color:#231f20;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:8pt;font-family:"Arial";font-style:italic}.c186{margin-left:28.8pt;padding-top:0.2pt;text-indent:-17.1pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:0.6pt}.c53{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Arial";font-style:italic}.c65{margin-left:16.8pt;padding-top:0pt;text-indent:-16.8pt;padding-bottom:0pt;line-height:0.9220961332321167;text-align:justify;margin-right:10.2pt}.c58{margin-left:26.1pt;padding-top:0.2pt;text-indent:-17pt;padding-bottom:0pt;line-height:0.9221321940422058;text-align:justify;margin-right:0.8pt}.c210{margin-left:11.7pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:0.5pt}.c144{padding-top:10.4pt;padding-bottom:0pt;line-height:1.015303373336792;text-indent:0.2pt;text-align:left;margin-right:8.4pt}.c93{padding-top:15.6pt;text-indent:-3.2pt;padding-bottom:0pt;line-height:1.0712363719940186;text-align:left;margin-right:103.2pt}.c105{margin-left:186.9pt;padding-top:21.4pt;padding-bottom:0pt;line-height:0.9221095442771912;text-indent:-186.9pt;text-align:left}.c86{margin-left:0pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:10.2pt}.c34{margin-left:0.4pt;padding-top:7.9pt;padding-bottom:0pt;line-height:0.8903119564056396;text-align:center;margin-right:7.4pt}.c287{padding-top:13.6pt;padding-bottom:0pt;line-height:1.0143470764160156;text-indent:0.1pt;text-align:justify;margin-right:7.1pt}.c289{padding-top:0.2pt;text-indent:-0pt;padding-bottom:0pt;line-height:0.922113835811615;text-align:left;margin-right:0.5pt}.c277{padding-top:0.5pt;text-indent:-0.4pt;padding-bottom:0pt;line-height:0.8363358378410339;text-align:left;margin-right:31.6pt}.c267{margin-left:0.1pt;padding-top:11.4pt;padding-bottom:0pt;line-height:0.968574047088623;text-align:justify;margin-right:8.4pt}.c116{margin-left:186.6pt;padding-top:1.4pt;padding-bottom:0pt;line-height:0.9221095442771912;text-indent:-186.6pt;text-align:left}.c240{padding-top:0.7pt;padding-bottom:0pt;line-height:0.9963939189910889;text-indent:10.1pt;text-align:justify;margin-right:8.1pt}.c268{padding-top:0.4pt;padding-bottom:0pt;line-height:1.0152268409729004;text-indent:9.5pt;text-align:justify;margin-right:8pt}.c59{margin-left:0pt;padding-top:0pt;padding-bottom:0pt;line-height:0.9220960736274719;text-align:left;margin-right:10.2pt}.c241{padding-top:11.3pt;text-indent:-0.1pt;padding-bottom:0pt;line-height:1.0143468379974365;text-align:justify;margin-right:8.5pt}.c80{margin-left:13.4pt;padding-top:21.1pt;padding-bottom:0pt;line-height:1.0137923955917358;text-align:center;margin-right:42.3pt}.c153{padding-top:12.4pt;padding-bottom:0pt;line-height:1.0143123865127563;text-indent:0.1pt;text-align:justify;margin-right:7.1pt}.c261{padding-top:0.5pt;text-indent:9.8pt;padding-bottom:0pt;line-height:0.9549053907394409;text-align:justify;margin-right:6.9pt}.c258{margin-left:25.9pt;padding-top:0.3pt;padding-bottom:0pt;line-height:1.0146238803863525;text-align:left;margin-right:42.4pt}.c154{margin-left:187pt;padding-top:0.5pt;padding-bottom:0pt;line-height:0.9221095442771912;text-indent:-187pt;text-align:left}.c217{padding-top:0pt;padding-bottom:0pt;line-height:0.9220964312553406;text-indent:0pt;text-align:left;margin-right:9.5pt}.c146{margin-left:11.7pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220961332321167;text-align:left;margin-right:0.3pt}.c14{margin-left:5.1pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:center;margin-right:1.1pt}.c88{margin-left:9.1pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:1.1pt}.c36{padding-top:0.3pt;padding-bottom:0pt;line-height:1.0155589580535889;text-indent:9.9pt;text-align:justify;margin-right:8.1pt}.c279{padding-top:0pt;text-indent:0.7pt;padding-bottom:0pt;line-height:0.9993377923965454;text-align:justify;margin-right:66.4pt}.c157{margin-left:15.6pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:0pt}.c172{margin-left:11.7pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220961332321167;text-align:center;margin-right:0.5pt}.c56{margin-left:187.3pt;padding-top:1pt;padding-bottom:0pt;line-height:0.9221095442771912;text-indent:-187.3pt;text-align:left}.c90{padding-top:7pt;padding-bottom:0pt;line-height:1.0153149366378784;text-indent:0.1pt;text-align:justify;margin-right:8.1pt}.c45{margin-left:0pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:left;margin-right:10.2pt}.c292{margin-left:3.9pt;padding-top:30.9pt;padding-bottom:0pt;line-height:1.0148215293884277;text-align:justify;margin-right:65pt}.c108{margin-left:9.1pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220964312553406;text-align:left;margin-right:1pt}.c136{padding-top:0.2pt;text-indent:-0pt;padding-bottom:0pt;line-height:0.9218241572380066;text-align:left;margin-right:10.2pt}.c226{padding-top:0pt;text-indent:0.8pt;padding-bottom:0pt;line-height:0.8363358378410339;text-align:left;margin-right:47.3pt}.c269{padding-top:6.6pt;text-indent:-0pt;padding-bottom:0pt;line-height:0.8942395448684692;text-align:justify;margin-right:7.1pt}.c100{padding-top:11.4pt;padding-bottom:0pt;line-height:1.0158501863479614;text-indent:0.1pt;text-align:justify;margin-right:8.1pt}.c179{margin-left:0.1pt;padding-top:11.4pt;padding-bottom:0pt;line-height:1.015249490737915;text-align:justify;margin-right:8pt}.c272{padding-top:11.4pt;padding-bottom:0pt;line-height:1.016735315322876;text-indent:0.6pt;text-align:justify;margin-right:8.2pt}.c247{margin-left:11.7pt;padding-top:0.2pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:left;margin-right:0pt}.c206{margin-left:5.1pt;padding-top:0pt;padding-bottom:0pt;line-height:0.922096848487854;text-align:center;margin-right:1.1pt}.c204{padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220964312553406;text-align:left;margin-right:1.1pt}.c40{margin-left:13.1pt;padding-top:18pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c61{padding-top:0pt;padding-bottom:0pt;line-height:1.0161181688308716;text-align:justify;margin-right:8.1pt}.c71{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:227pt}.c72{margin-left:6.1pt;padding-top:14.7pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c113{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:1.5pt}.c140{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:190.1pt}.c26{margin-left:0.3pt;padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c83{margin-left:2.8pt;padding-top:568.2pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c237{margin-left:2.8pt;padding-top:26.8pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c271{margin-left:2.8pt;padding-top:22.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c193{margin-left:1pt;padding-top:3.9pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c39{margin-left:0pt;padding-top:0.2pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c211{padding-top:5.5pt;padding-bottom:0pt;line-height:0.885044515132904;text-align:justify;margin-right:7.2pt}.c167{margin-left:3.9pt;padding-top:1.2pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c126{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:224.7pt}.c23{margin-left:28.7pt;padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c174{padding-top:10.9pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:7.4pt}.c182{padding-top:0.2pt;padding-bottom:0pt;line-height:0.9221208691596985;text-align:left;margin-right:8.2pt}.c79{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:65pt}.c130{margin-left:1pt;padding-top:18.9pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c82{margin-left:6.4pt;padding-top:7pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c125{padding-top:0.2pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:14.7pt}.c122{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:242.5pt}.c196{margin-left:6.3pt;padding-top:21.8pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c66{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:0.7pt}.c48{margin-left:6.4pt;padding-top:0.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c195{margin-left:3.1pt;padding-top:18.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c260{padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:0.6pt}.c44{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:1.4pt}.c155{margin-left:6.4pt;padding-top:13.4pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c213{margin-left:0.1pt;padding-top:22.6pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c201{margin-left:26pt;padding-top:0.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c168{margin-left:6.3pt;padding-top:7pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c231{padding-top:11pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:63.9pt}.c77{margin-left:12.9pt;padding-top:0.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c156{margin-left:2.9pt;padding-top:13.4pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c96{margin-left:6.5pt;padding-top:24pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c296{padding-top:0pt;padding-bottom:0pt;line-height:0.9220961332321167;text-align:left;margin-right:10.2pt}.c283{margin-left:2.8pt;padding-top:21.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c238{margin-left:0.1pt;padding-top:12.8pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c54{margin-left:0.3pt;padding-top:19.7pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c24{margin-left:2.8pt;padding-top:30.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c227{padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220964312553406;text-align:left;margin-right:10pt}.c97{margin-left:0.3pt;padding-top:13.9pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c74{margin-left:0.4pt;padding-top:0.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c266{margin-left:6.3pt;padding-top:13pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c41{margin-left:0.2pt;padding-top:22.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c251{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:0.8pt}.c257{padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220983386039734;text-align:center;margin-right:1.1pt}.c151{margin-left:6.5pt;padding-top:11.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c184{margin-left:2.8pt;padding-top:49.4pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c189{margin-left:6.3pt;padding-top:12.9pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c7{margin-left:0.7pt;padding-top:5.4pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c120{padding-top:11.4pt;padding-bottom:0pt;line-height:1.0177333354949951;text-align:center;margin-right:7.3pt}.c46{margin-left:1pt;padding-top:21.4pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c228{margin-left:0.2pt;padding-top:22.7pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c135{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:7.4pt}.c281{padding-top:5.5pt;padding-bottom:0pt;line-height:0.8758798241615295;text-align:justify;margin-right:7.2pt}.c169{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:67.8pt}.c236{padding-top:0.2pt;padding-bottom:0pt;line-height:0.9220954775810242;text-align:left;margin-right:0.8pt}.c49{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:241.8pt}.c188{margin-left:0.9pt;padding-top:19.8pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c35{padding-top:6.9pt;padding-bottom:0pt;line-height:1.0;text-align:right;margin-right:37.2pt}.c180{padding-top:32.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c253{padding-top:18.7pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c244{padding-top:1.1pt;padding-bottom:0pt;line-height:0.9212735295295715;text-align:left}.c248{padding-top:2.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c91{padding-top:22.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c158{padding-top:15.1pt;padding-bottom:0pt;line-height:0.8363386392593384;text-align:left}.c76{padding-top:410.1pt;padding-bottom:0pt;line-height:0.8759387135505676;text-align:left}.c152{padding-top:21.2pt;padding-bottom:0pt;line-height:0.8330000042915344;text-align:left}.c197{padding-top:21pt;padding-bottom:0pt;line-height:0.9221095442771912;text-align:left}.c275{padding-top:24pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c70{padding-top:0pt;padding-bottom:0pt;line-height:1.6256577968597412;text-align:left}.c270{padding-top:15.1pt;padding-bottom:0pt;line-height:0.8330000042915344;text-align:left}.c29{padding-top:0pt;padding-bottom:0pt;line-height:0.8330000042915344;text-align:left}.c185{padding-top:1.1pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c73{padding-top:0.2pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c145{padding-top:13.1pt;padding-bottom:0pt;line-height:0.8363217115402222;text-align:left}.c262{padding-top:1.4pt;padding-bottom:0pt;line-height:0.8703663945198059;text-align:left}.c295{padding-top:1.4pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c106{padding-top:0.6pt;padding-bottom:0pt;line-height:1.0165202617645264;text-align:left}.c221{padding-top:21.2pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c225{padding-top:15.1pt;padding-bottom:0pt;line-height:0.8363348245620728;text-align:left}.c160{padding-top:70pt;padding-bottom:0pt;line-height:0.8330000042915344;text-align:left}.c191{padding-top:5.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c110{padding-top:11.4pt;padding-bottom:0pt;line-height:0.9221095442771912;text-align:left}.c207{padding-top:615.3pt;padding-bottom:0pt;line-height:0.8760836720466614;text-align:left}.c22{padding-top:1.7pt;padding-bottom:0pt;line-height:0.9221096634864807;text-align:left}.c243{padding-top:11.6pt;padding-bottom:0pt;line-height:0.8354856371879578;text-align:left}.c112{padding-top:20.7pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c28{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c259{padding-top:0.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c285{padding-top:44.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c209{padding-top:348.4pt;padding-bottom:0pt;line-height:0.8762865662574768;text-align:left}.c177{padding-top:13.1pt;padding-bottom:0pt;line-height:1.1121129989624023;text-align:left}.c148{padding-top:7pt;padding-bottom:0pt;line-height:0.970716655254364;text-align:left}.c139{padding-top:0pt;padding-bottom:0pt;line-height:0.9221103191375732;text-align:left}.c133{padding-top:33.7pt;padding-bottom:0pt;line-height:1.110682487487793;text-align:left}.c162{padding-top:1.5pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c81{padding-top:1pt;padding-bottom:0pt;line-height:0.9221095442771912;text-align:left}.c176{padding-top:0.2pt;padding-bottom:0pt;line-height:1.0;text-align:right}.c107{padding-top:557.1pt;padding-bottom:0pt;line-height:0.8759617209434509;text-align:left}.c21{padding-top:29.6pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c274{padding-top:20.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c166{padding-top:27.3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c138{padding-top:11pt;padding-bottom:0pt;line-height:0.9221095442771912;text-align:left}.c164{margin-left:187pt;text-indent:-187pt;margin-right:8.4pt}.c254{background-color:#ffffff;max-width:535.5pt;padding:10.6pt 11.6pt 18.2pt 47.9pt}.c219{margin-left:187.7pt;text-indent:-187.7pt}.c233{text-indent:0pt}.c252{margin-left:11.7pt}.c235{margin-left:0.4pt}.c127{margin-left:7.2pt}.c78{margin-left:21.8pt}.c84{margin-left:0.8pt}.c134{margin-left:0pt}.c239{margin-left:0.2pt}.c224{margin-left:29pt}.c147{margin-left:9.1pt}.c38{margin-left:3.2pt}.c293{margin-left:22pt}.c203{margin-left:29.4pt}.c276{margin-right:13.2pt}.c50{margin-left:3.6pt}.c280{margin-left:22.4pt}.c290{margin-left:5.1pt}.title{padding-top:24pt;color:#000000;font-weight:700;font-size:36pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:18pt;color:#666666;font-size:24pt;padding-bottom:4pt;font-family:"Georgia";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:24pt;color:#000000;font-weight:700;font-size:24pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-weight:700;font-size:18pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:14pt;color:#000000;font-weight:700;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:12pt;color:#000000;font-weight:700;font-size:12pt;padding-bottom:2pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:11pt;color:#000000;font-weight:700;font-size:11pt;padding-bottom:2pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:10pt;color:#000000;font-weight:700;font-size:10pt;padding-bottom:2pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}</style></head><body class="c254 doc-content"><p class="c28 c50"><span class="c250">RESEARCH ARTICLE </span></p><p class="c35"><span class="c18">www.advancedscience.com </span></p><p class="c156"><span class="c141">An AI Agent for Fully Automated Multi-Omic Analyses </span></p><p class="c50 c93"><span class="c178">Juexiao Zhou, Bin Zhang, Guowei Li, Xiuying Chen, Haoyang Li, Xiaopeng Xu, Siyuan Chen, Wenjia He, Chencheng Xu, Liwei Liu,* and Xin Gao* </span></p><p class="c133"><span class="c64">With the fast-growing and evolving omics data, the demand for streamlined and adaptable tools to handle bioinformatics analysis continues to grow. In response to this need, Automated Bioinformatics Analysis (AutoBA) is introduced, an autonomous AI agent designed explicitly for fully automated multi-omic analyses based on large language models (LLMs). AutoBA simplifies the analytical process by requiring minimal user input while delivering detailed step-by-step plans for various bioinformatics tasks. AutoBA&rsquo;s unique capacity to self-design analysis processes based on input data variations further underscores its versatility. Compared with online bioinformatic services, AutoBA offers multiple LLM backends, with options for both online and local usage, prioritizing data security and user privacy. In comparison to ChatGPT and open-source LLMs, an automated code repair (ACR) mechanism in AutoBA is designed to improve its stability in automated end-to-end bioinformatics analysis tasks. Moreover, different from the predefined pipeline, AutoBA has adaptability in sync with emerging bioinformatics tools. Overall, AutoBA represents an advanced and convenient tool, offering robustness and adaptability for conventional multi-omic analyses. </span></p><p class="c28"><span class="c16">1. Introduction </span></p><p class="c148"><span class="c12">Bioinformatics is an interdisciplinary field that encompasses computational, statistical, and biological approaches to analyze, understand and interpret com plex biological data.</span><span class="c3">[</span><span class="c13">1&ndash;3</span><span class="c3">] </span><span class="c12">With the rapid growth of gigabyte-sized biological data generated from various high-throughput technologies, bioinformatics has become an essential tool for researchers to make sense of these massive datasets and extract meaningful biological insights. The appli cations of bioinformatics typically cover diverse fields such as genome analysis,</span><span class="c3">[</span><span class="c13">4,5</span><span class="c3">]</span><span class="c68">&nbsp;</span><span class="c12">structural bioinformatics,</span><span class="c3">[</span><span class="c13">6&ndash;7</span><span class="c3">] </span><span class="c12">systems biology,</span><span class="c3">[</span><span class="c13">8</span><span class="c3">] </span><span class="c12">data and text mining,</span><span class="c3">[</span><span class="c13">9&ndash;10</span><span class="c3">]</span><span class="c68">&nbsp;</span><span class="c12">phylogenetics,</span><span class="c3">[</span><span class="c13">11&ndash;12</span><span class="c3">] </span><span class="c12">and population analysis,</span><span class="c3">[</span><span class="c13">13</span><span class="c3">] </span><span class="c12">which has further enabled significant advances in personalized medicine</span><span class="c3">[</span><span class="c13">14</span><span class="c3">] </span><span class="c12">and drug discovery.</span><span class="c3">[</span><span class="c13">15</span><span class="c3">]</span><span class="c68">&nbsp;</span></p><p class="c106"><span class="c12">In broad terms, bioinformatics could be categorized into two primary do mains: the development of innovative </span></p><p class="c192"><span class="c0">J. Zhou, B. Zhang, X. Chen, H. Li, X. Xu, S. Chen, W. He, C. Xu, X. Gao Computer Science Program </span></p><p class="c47"><span class="c0">Computer, Electrical and Mathematical Sciences and Engineering Division </span></p><p class="c134 c226"><span class="c0">King Abdullah University of Science and Technology (KAUST) Thuwal 23955-6900, Kingdom of Saudi Arabia </span></p><p class="c28 c84"><span class="c0">E-mail: </span><span class="c31">xin.gao@kaust.edu.sa </span></p><p class="c235 c277"><span class="c0">J. Zhou, B. Zhang, X. Chen, H. Li, X. Xu, S. Chen, W. He, C. Xu, X. Gao Center of Excellence on Smart Health </span></p><p class="c28 c84"><span class="c0">King Abdullah University of Science and Technology </span></p><p class="c28 c134"><span class="c0">Thuwal 23955-6900, Kingdom of Saudi Arabia </span></p><p class="c74"><span class="c0">G. Li </span></p><p class="c28 c84"><span class="c0">Laboratory of Health Intelligence </span></p><p class="c28 c84"><span class="c0">Huawei Technologies Co., Ltd </span></p><p class="c28 c235"><span class="c0">Shenzhen 210000, China </span></p><p class="c84 c259"><span class="c0">L. Liu </span></p><p class="c28 c134"><span class="c0">Advanced Computing and Storage Laboratory </span></p><p class="c28 c235"><span class="c0">Central Research Institute </span></p><p class="c26"><span class="c0">2012 Laboratories, Huawei Technologies Co., Ltd </span></p><p class="c28 c84"><span class="c0">Nanjing, Jiangsu 210000, China </span></p><p class="c28 c84"><span class="c0">E-mail: </span><span class="c31">liuliwei5@huawei.com </span></p><p class="c34"><span class="c0">The ORCID identification number(s) for the author(s) of this article can be found under </span><span class="c31">https://doi.org/10.1002/advs.202407094 </span><span class="c0">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH. This is an open access article under the terms of the </span><span class="c31">Creative Commons Attribution </span><span class="c0">License, which permits use, distribution and reproduction in any medium, provided the original work is properly cited. </span></p><p class="c7"><span class="c30">DOI: 10.1002/advs.202407094 </span></p><p class="c222"><span class="c12">algorithms to address various biological challenges,</span><span class="c3">[</span><span class="c13">16&ndash;20</span><span class="c3">] </span><span class="c12">and the application of established tools to analyze extensive bi ological datasets,</span><span class="c3">[</span><span class="c13">21,22</span><span class="c3">] </span><span class="c12">especially high-throughput sequencing data. Developing new bioinformatics software requires a sub stantial grasp of biology and programming expertise. Along side the development of novel computational methods, one of the most prevalent applications of bioinformatics is the investigation of biological data using the existing tools and pipelines,</span><span class="c3">[</span><span class="c13">23,24</span><span class="c3">] </span><span class="c12">which typically involves a sequential, flow-based analysis of omics data, encompassing variety types of datasets like whole genome sequencing (WGS),</span><span class="c3">[</span><span class="c13">25</span><span class="c3">] </span><span class="c12">whole exome sequenc ing (WES), RNA sequencing (RNA-seq),</span><span class="c3">[</span><span class="c13">26</span><span class="c3">] </span><span class="c12">single-cell RNA-seq (scRNA-Seq),</span><span class="c3">[</span><span class="c13">27</span><span class="c3">] </span><span class="c12">transposase-accessible chromatin with sequenc ing (ATAC-Seq),</span><span class="c3">[</span><span class="c13">28</span><span class="c3">] </span><span class="c12">ChIP-seq,</span><span class="c3">[</span><span class="c13">29</span><span class="c3">] </span><span class="c12">and spatial transcriptomics.</span><span class="c3">[</span><span class="c13">30</span><span class="c3">]</span><span class="c68">&nbsp;</span></p><p class="c265"><span class="c12">For example, the conventional analytical framework for bulk RNA-seq involves a meticulously structured sequence of com putational steps.</span><span class="c3">[</span><span class="c13">31</span><span class="c3">] </span><span class="c12">This intricate pipeline reveals its com plexity through a series of carefully orchestrated stages. It begins with quality control,</span><span class="c3">[</span><span class="c13">32</span><span class="c3">] </span><span class="c12">progresses to tasks such as adapter trimming</span><span class="c3">[</span><span class="c13">33</span><span class="c3">] </span><span class="c12">and the removal of low-quality reads, and then moves on to critical steps like genome or transcriptome alignment.</span><span class="c3">[</span><span class="c13">34</span><span class="c3">] </span><span class="c12">Furthermore, it extends to some advanced tasks, including the identification of splice junctions,</span><span class="c3">[</span><span class="c13">35</span><span class="c3">] </span><span class="c12">quantifica tion through read counting,</span><span class="c3">[</span><span class="c13">36</span><span class="c3">] </span><span class="c12">and the rigorous examination of differential gene expression.</span><span class="c3">[</span><span class="c13">37</span><span class="c3">] </span><span class="c12">Moreover, the pipeline delves into the intricate domain of alternative splicing</span><span class="c3">[</span><span class="c13">38</span><span class="c3">] </span><span class="c12">and isoform </span></p><p class="c237"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (1 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH</span></p><p class="c70"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c62"><span class="c12">analysis.</span><span class="c3">[</span><span class="c13">39</span><span class="c3">] </span><span class="c12">This progressive journey ultimately ends in down stream tasks like the exploration of functional enrichment,</span><span class="c3">[</span><span class="c13">40</span><span class="c3">]</span><span class="c68">&nbsp;</span><span class="c12">providing a comprehensive range of analytical pursuits. Com pared to bulk RNA-seq, ChIP-seq involves distinct down stream tasks, such as peak calling,</span><span class="c3">[</span><span class="c13">41</span><span class="c3">] </span><span class="c12">motif discovery,</span><span class="c3">[</span><span class="c13">42</span><span class="c3">] </span><span class="c12">peak annotation</span><span class="c3">[</span><span class="c13">43</span><span class="c3">] </span><span class="c12">and so on. In summary, the analysis of different types of omics data requires professional skills and a comprehen sive comprehension of the corresponding field, particularly for customized data analysis. Moreover, the methods and pipelines might vary across different bioinformaticians and they even may evolve with the development of more advanced algorithms. </span></p><p class="c239 c261"><span class="c12">Meanwhile, online semi-automatic bioinformatics analysis platforms are currently in vogue,</span><span class="c3">[</span><span class="c13">44</span><span class="c3">] </span><span class="c12">such as iDEP,</span><span class="c3">[</span><span class="c13">45</span><span class="c3">] </span><span class="c12">ICARUS</span><span class="c3">[</span><span class="c13">46</span><span class="c3">]</span><span class="c68">&nbsp;</span><span class="c12">and STellaris.</span><span class="c3">[</span><span class="c13">47</span><span class="c3">] </span><span class="c12">However, they often necessitate the uploading of either raw data or pre-processed statistics by users, which could potentially give rise to additional privacy concerns and data leak age risks.</span><span class="c3">[</span><span class="c13">48</span><span class="c3">]</span><span class="c68">&nbsp;</span></p><p class="c240"><span class="c12">In the context described above, the bioinformatics commu nity grapples with essential concerns regarding the standardiza tion, portability, and reproducibility of analysis pipelines.</span><span class="c3">[</span><span class="c13">49&ndash;51</span><span class="c3">]</span><span class="c68">&nbsp;</span><span class="c12">Moreover, achieving proficiency in utilizing these pipelines for data analysis demands additional training, posing challenges for many wet lab researchers due to its potential complexity and time-consuming nature. Even dry-lab researchers may find the repetitive process of running and debugging these pipelines to be quite tedious.</span><span class="c3">[</span><span class="c13">52</span><span class="c3">] </span><span class="c12">Meanwhile, bioinformatics data analysis train ing incurs substantial costs. The elevated expenses associated with training in bioinformatics data analysis could be attributed to the highly specialized nature of the field, the need for multi modal data analysis, the evolution of technologies, restricted computing resources, the expense of training materials and tools, as well as the operational costs of training institutions. These factors collectively contribute to the high cost of bioinformatics training.</span><span class="c3">[</span><span class="c13">53</span><span class="c3">] </span><span class="c12">Consequently, there is a growing anticipation within the community for the development of a more user-friendly, low code, multi-functional, automated, and natural language-driven intelligent tool tailored for end-to-end bioinformatics analysis. Such a tool has the potential to generate significant excitement and benefit researchers across the field. </span></p><p class="c291"><span class="c12">Over the past few months, the rapid advancement of Large Language Models (LLMs)</span><span class="c3">[</span><span class="c13">54</span><span class="c3">] </span><span class="c12">has raised substantial expectations for the enhancement of scientific research, particularly in the field of biology.</span><span class="c3">[</span><span class="c13">55&ndash;57</span><span class="c3">] </span><span class="c12">These advancements hold promise for applications such as disease diagnosis,</span><span class="c3">[</span><span class="c13">58&ndash;61</span><span class="c3">] </span><span class="c12">drug discovery,</span><span class="c3">[</span><span class="c13">62</span><span class="c3">]</span><span class="c68">&nbsp;</span><span class="c12">and all. In the realm of bioinformatics, LLMs, such as ChatGPT, also demonstrate immense potential in tasks related to bioinfor matics education</span><span class="c3">[</span><span class="c13">63</span><span class="c3">] </span><span class="c12">and code generation.</span><span class="c3">[</span><span class="c13">64</span><span class="c3">] </span><span class="c12">While researchers have found ChatGPT to be a valuable tool in facilitating bioin formatics research, such as data analysis, there remains a strong requirement for human intervention in the execution process. ChatGPT shows sensitivity to the nuances of user queries, resulting in diverse responses based on the prompts, which is the reason why prompt engineering is getting huge attention.</span><span class="c3">[</span><span class="c13">65</span><span class="c3">]</span><span class="c68">&nbsp;</span><span class="c12">Given the specialized nature of bioinformatics tools, ChatGPT is also susceptible to potential issues, such as misinterpreting pa rameters, errors in software utilization, and other bugs that may arise during code generation. Users may encounter the necessity for ongoing engagement with ChatGPT, involving a continuous cycle of inquiry, code generation, execution, and debugging to en </span></p><p class="c89"><span class="c12">sure desired performance. AutoGPT,</span><span class="c3">[</span><span class="c13">66</span><span class="c3">] </span><span class="c12">as a recently developed, advanced, and experimental open-source autonomous AI agent, has the capacity to string together LLM-generated &ldquo;thoughts&rdquo; to autonomously achieve user-defined objectives. Nevertheless, given the intricate and specialized nature of bioinformatics tasks, such as specialized software, the direct application of AutoGPT in this field still presents significant challenges. Notably, it faces difficulties in effectively managing the intricate software requirements of bioinformatics, encompassing tasks such as installation, software calls, and parameter settings. </span></p><p class="c123"><span class="c12">In this study, we introduce Automated Bioinformatics Analysis (AutoBA), an autonomous AI agent tailored for comprehensive and conventional multi-omic analyses, as it can be applied to the analysis of different omics datasets. AutoBA simplifies user inter actions to just three inputs: data path, data description, and the final objective. This tool autonomously proposes analysis plans, generates code, executes codes, and conducts subsequent data analysis by using our well-designed prompts. We implemented AutoBA as open-source software that offers multiple LLM back ends, with options for both online and local usage, prioritizing data security and user privacy (</span><span class="c30">Figure </span><span class="c18">1</span><span class="c12">). To show the reliability of AutoBA, we tested it in a large number of real-world multi-omic analysis scenarios (</span><span class="c30">Figure </span><span class="c18">2</span><span class="c12">). AutoBA, serving as an AI agent tai lored for bioinformatics data analysis, could address the surging demand for streamlined multi-omics data analysis, mitigate the financial challenges associated with bioinformatics training, and cater to diverse customization requirements. Compared with on line bioinformatic services, AutoBA offers multiple LLM back ends, with options for both online and local usage, prioritizing data security and user privacy (</span><span class="c30">Table </span><span class="c18">1</span><span class="c12">). In comparison to Chat GPT and open-source LLMs, we have designed an automated code repair mechanism in AutoBA to improve its stability in au tomated end-to-end bioinformatics analysis tasks. Moreover, dif ferent from the predefined pipeline, AutoBA has adaptability in sync with emerging bioinformatics tools. In summary, AutoBA is the first agent of this kind and represents a significant leap in the application of LLMs and automated AI agents within the do main of bioinformatics, highlighting their potential to accelerate future research in this field. </span></p><p class="c266"><span class="c16">2. Experimental Section </span></p><p class="c168"><span class="c30">2.1. The Overall Framework Design of AutoBA </span></p><p class="c52"><span class="c12">AutoBA is the first autonomous AI agent tailor-made for con ventional multi-omic analyses. As illustrated in Figure </span><span class="c10">1</span><span class="c12">, con ventional bioinformatics typically entails the use of pipelines to analyze diverse data types such as WGS, WES, RNA-seq, single cell RNA-seq, ChIP-seq, ATAC-seq, spatial transcriptomics, and more, all requiring the utilization of various tools. Users are tradi tionally tasked with selecting the appropriate tools based on their specific analysis needs. In practice, this process involves config uring the environment, installing software, writing code, and de bugging, which are time-consuming and labor-intensive. </span></p><p class="c249"><span class="c12">With the advent of AutoBA, this labor-intensive process is revolutionized. Users are relieved from the burden of dealing with multiple software packages and need only pro vide three key inputs in YAML format: the data path (e.g., /data/SRR1374921.fasta.gz), data description (e.g., single-end </span></p><p class="c221"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (2 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c29"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 339.37px; height: 253.77px;"><img alt="" src="images/image2.png" style="width: 339.37px; height: 253.77px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span></p><p class="c29"><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c76"><span class="c2">Figure 1. </span><span class="c0">Design of AutoBA. AutoBA stands as the first autonomous AI agent meticulously crafted for conventional multi-omic analyses. Remarkably user-friendly, AutoBA simplifies the analytical process by requiring minimal user input, including data path, data description, and the final objective, while delivering detailed step-by-step plans for various bioinformatics tasks. With these inputs, it autonomously proposes analysis plans, generates code, executes codes, and conducts subsequent data analysis by using our well-designed prompts. AutoBA was implemented as open-source software that offers multiple LLM backends, with options for both online and local deployment, prioritizing data security and user privacy and offering a streamlined and efficient solution for bioinformatics tasks. Step 1 and Step 3 require human intervention, while Step 2 requires no human intervention. Due to the numerous and complex nature of these specific items, they are represented with ellipsis to indicate the vast and detailed possibilities that cannot be fully enumerated in this limited space. </span></p><p class="c153"><span class="c12">reads in condition A), and the ultimate analysis goal (e.g., identify differentially expressed genes). AutoBA takes over by autonomously analyzing the data, generating comprehensive step-by-step plans, composing code for each step, executing the generated code, and conducting in-depth analysis. Depending on the complexity and difficulty of the tasks, users can expect AutoBA to complete the tasks within a matter of minutes to a few hours, all without the need for additional human intervention (</span><span class="c30">Table </span><span class="c18">2 </span><span class="c12">and Figure </span><span class="c10">2</span><span class="c12">). </span></p><p class="c213"><span class="c30">2.2. Prompt Engineering of AutoBA </span></p><p class="c120"><span class="c12">To initiate AutoBA, users provide three essential inputs: the data path, data description, and the previously mentioned analysis ob </span></p><p class="c198"><span class="c12">jective. AutoBA comprises three distinct phases: the planning phase, the code generation phase, and the execution phase as shown in Step 2 of Figure </span><span class="c10">1</span><span class="c12">. During the planning phase, Au toBA meticulously outlines a comprehensive step-by-step anal ysis plan. This plan includes details such as the software name and version to be used at each step, along with guided actions and specific sub-tasks for each stage. Subsequently, in the code generation phase, AutoBA systematically follows the plan and generates codes for sub-tasks, which entails procedures like con figuring the environment, installing the necessary software, and writing code. Then, in the execution phase, AutoBA executes the generated code. In light of this workflow, AutoBA incorporates two distinct prompts: one tailored for the planning phase and the other for the code generation phase. Intensive experiments have shown that these two sets of prompts are essential for the </span></p><p class="c283"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (3 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c29"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 341.24px; height: 293.25px;"><img alt="" src="images/image3.png" style="width: 341.24px; height: 293.25px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span></p><p class="c29"><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c209"><span class="c2">Figure 2. </span><span class="c0">Method design and evaluation of AutoBA. a) AutoBA workflow design and technical details. b) Pie chart indicates the number of all cases used for validating AutoBA. </span></p><p class="c144"><span class="c12">proper functioning of AutoBA in automated bioinformatics anal ysis tasks. </span></p><p class="c218"><span class="c12">The prompts for both the planning phase and the code gen eration phase are displayed in the Supporting Information. In both prompt designs, the term </span><span class="c53">blacklist </span><span class="c12">pertains to the user&rsquo;s per sonalized list of prohibited software. The current default black list contains several tools that frequently caused errors during the testing processes. Meanwhile, </span><span class="c53">data list </span><span class="c12">encompasses the in puts necessary for AutoBA, encompassing data paths and data descriptions. The term </span><span class="c53">current goal </span><span class="c12">serves as the final objective </span></p><p class="c173"><span class="c12">during the planning phase and as the sub-goal in the execution phase, while </span><span class="c53">history summary </span><span class="c12">encapsulates AutoBA&rsquo;s memory of previous actions and information. </span></p><p class="c196"><span class="c30">2.3. Memory Management of AutoBA </span></p><p class="c245"><span class="c12">A memory mechanism is incorporated within AutoBA to enable it to generate code more effectively by drawing from past actions, thus avoiding unnecessary repetition of certain steps. AutoBA </span></p><p class="c32"><span class="c2">Table 1. </span><span class="c0">Qualitative comparison of AutoBA against other methods. All methods were conceptually assessed with seven metrics, including user friendliness, time efficiency, diminished human intervention (degree of automation), ease of redevelopment, generality, robustness, and privacy con siderations. * denotes a category of methods rather than a specific one. For instance, Online Webserver refers to platforms like iDEP and ICARUS, and Open Source LLMs includes models such as Llama2 and CodeLlama. </span></p><p class="c121"><span class="c5">Methods Easy to Master Save Time Reduced Human Intervention </span></p><p class="c113"><span class="c5">Redevelopment Generalizability Robustness Privacy </span></p><p class="c238"><span class="c5">AutoBA &#10003;&#10003;&#10003; &#10003;&#10003;&#10003; &#10003;&#10003;&#10003; &#10003;&#10003;&#10003; &#10003;&#10003;&#10003; &#10003;&#10003;&#10003; &#10003;&#10003;&#10003; </span></p><p class="c162"><span class="c5">Conventional </span></p><p class="c28"><span class="c5">Bioinformatics Tools </span></p><p class="c28"><span class="c5">&#10003; &#10003; &ndash; &#10003;&#10003; &#10003; &#10003;&#10003; &#10003;&#10003;&#10003; </span></p><p class="c232"><span class="c5">Online Webserver</span><span class="c15">&lowast; </span><span class="c5">&#10003;&#10003; &#10003;&#10003; &ndash; &ndash; &#10003; &#10003; &ndash; AutoGPT &#10003;&#10003;&#10003; &#10003;&#10003; &#10003;&#10003; &#10003;&#10003; &#10003;&#10003;&#10003; &#10003; &#10003;&#10003;&#10003; ChatGPT &#10003;&#10003;&#10003; &#10003;&#10003; &#10003; &#10003;&#10003; &#10003;&#10003;&#10003; &#10003; &#10003;&#10003; Open source LLMs</span><span class="c15">&lowast; </span><span class="c5">&#10003; &#10003;&#10003; &#10003; &#10003;&#10003; &#10003;&#10003;&#10003; &#10003; &#10003;&#10003;&#10003; </span></p><p class="c134 c166"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (4 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c29"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span></p><p class="c29"><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c158"><span class="c2">Table 2. </span><span class="c0">Summary of AutoBA application scenarios in bioinformatics multi-omics analysis. The table displays a comprehensive list of 40 real-world cases utilized to assess AutoBA, providing information on the class of the cases, the respective task name, and the corresponding case ID. </span></p><p class="c177"><span class="c5">Bioinformatics Pipelines Tasks Types of Omics Case ID WGS data analysis Genome assembly Genomics 1.1 WGS/WES data analysis Somatic SNV+indel calling Genomics 2.1 WGS/WES data analysis Somatic SNV+indel calling and annotation Genomics 2.2 WGS/WES data analysis Structure variation identification with normal Genomics 2.3 WGS/WES data analysis Structure variation identification without normal Genomics 2.4 ChIP-seq data analysis Peak calling Genomics 3.1 ChIP-seq data analysis Motif discovery for binding sites Genomics 3.2 ChIP-seq data analysis Functional enrichment of target gene Genomics 3.3 Bisulfite-Seq data analysis Identifying DNA methylation Genomics 4.1 ATAC-seq data analysis Identifying open chromatin regions Genomics 5.1 DNase-seq data analysis Identifying Dnasel hypersensitive site Genomics 6.1 4C-seq data analysis Find genomics interactions Genomics 7.1 Nanopore DNA sequencing data analysis Genome assembly Genomics 8.1 Nanopore DNA sequencing data analysis Tandem repeats variation identification Genomics 8.2 PacBio DNA sequencing data analysis Genome assembly Genomics 9.1 RNA-Seq data analysis Find Differentially expressed genes Transcriptomics 10.1 RNA-Seq data analysis Identify the top5 downregulated genes Transcriptomics 10.2 RNA-Seq data analysis Predict Fusion gene with annotation Transcriptomics 10.3 RNA-Seq data analysis Isoform expression Transcriptomics 10.4 RNA-Seq data analysis Splicing analysis Transcriptomics 10.5 RNA-Seq data analysis APA analysis Transcriptomics 10.6 RNA-Seq data analysis RNA editing Transcriptomics 10.7 RNA-Seq data analysis Circular RNA identification Transcriptomics 10.8 Small RNA sequencing data analysis microRNA quantification Transcriptomics 11.1 Small RNA sequencing data analysis microRNA prediction Transcriptomics 11.2 CAGE-seq data analysis TSS identification Transcriptomics 12.1 3&rsquo; end-seq data analysis PAS (polyadenylation site) identification Transcriptomics 13.1 Nanopore RNA sequencing data analysis Isoform expression Transcriptomics 14.1 PacBio RNA sequencing data analysis Isoform expression Transcriptomics 15.1 CLIP-seq data analysis Identify protein-RNA crosslink sites Transcriptomics 16.1 RIP-seq data analysis Find enriched genes bounded by RBP Transcriptomics 16.2 Ribo-seq data analysis Identify translated ORFs Transcriptomics 17.1 single-cell RNA-seq data analysis Cell clustering from fastq data Transcriptomics 18.1 single-cell RNA-seq data analysis Find differentially expressed genes based on count matrix Transcriptomics 18.2 single-cell RNA-seq data analysis Find marker genes based on count matrix Transcriptomics 18.3 single-cell RNA-seq data analysis Cell clustering and visualization Transcriptomics 18.4 Spatial transcriptomics Neighborhood enrichment analysis Transcriptomics 19.1 Spatial transcriptomics Single-cell mapping Transcriptomics 19.2 Mass spectrometry data analysis Protein expression quantification Proteomics 20.1 Mass spectrometry data analysis Metabolites quantification Metabolomics 21.1 </span></p><p class="c287"><span class="c12">meticulously logs the outcome of each step in a specific format, and all these historical records become part of the input for the subsequent prompt. In the planning phase, memories are structured as follows: &ldquo;First, you provided input in the format &lsquo;file path: file description&rsquo; in a list: </span><span class="c53">&lt;data list&gt;</span><span class="c12">. You devised a detailed plan to accomplish your overarching objective. Your overarching goal is </span><span class="c53">&lt;global goal&gt;</span><span class="c12">. Your plan involves </span><span class="c53">&lt;tasks&gt;</span><span class="c12">.&rdquo; In the code generation phase, memories follow this format: &ldquo;Then, you successfully completed the task: </span><span class="c53">&lt;task&gt; </span><span class="c12">with the corresponding code: </span><span class="c53">&lt;code&gt;</span><span class="c12">.&rdquo; </span></p><p class="c28 c127"><span class="c30">2.4. Automatic Code Repair of AutoBA </span></p><p class="c55"><span class="c12">AutoBA incorporates an automatic code repair (ACR) module designed to streamline the debugging process and enhance the reliability of generated code. During the code execution phase, AutoBA identifies errors from the output stream called standard error (stderr) and standard output (stdout). Once an error is detected, these detected errors will be integrated into the prompt for code regeneration, ensuring a repetitive cycle until the generated code successfully executes without errors. </span></p><p class="c271"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (5 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c70"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c28 c239"><span class="c30">2.5. Evaluation of AutoBA </span></p><p class="c100"><span class="c12">The results produced by AutoBA undergo thorough validation by bioinformatics experts. This validation process encompasses a comprehensive review of the proposed plans, generated codes, execution of the code, and confirmation of the results for ac curacy and reliability. AutoBA&rsquo;s development and validation are built upon a specific environment and software stack, which in cludes Ubuntu version 20.04, Python 3.10.0, and openai version 0.27.6. These environment and software specifications form the robust foundation for AutoBA&rsquo;s functionality in the field of bioin formatics, ensuring its reliability and effectiveness. To further as sess the usability of AutoBA, a comparative analysis involving the following methods was conducted: 1) AutoBA (w/o ACR, online with ChatGPT-4), 2) AutoBA (with ACR, online with ChatGPT 4), 3) AutoBA (w/o ACR, offline with CodeLlama-34B-Instruct), 4) AutoBA (with ACR, offline with CodeLlama-34B-Instruct), 5) AutoGPT, 6) ChatGPT-3.5, 7) ChatGPT-4 and 8) CodeLlama-34B Instruct. Given that prompt engineering and workflow design is a distinctive innovation of AutoBA, during the evaluation of Au toGPT, ChatGPT-3.5, ChatGPT-4, and CodeLlama-34B-Instruct, user behavior was emulated by utilizing a generalized and uni form prompt as shown in the supplementary information. </span></p><p class="c41"><span class="c30">2.6. Online and Local LLM Backends of AutoBA </span></p><p class="c267"><span class="c12">AutoBA offers several versions of LLM backends, including on line backends based on ChatGPT-3.5 and ChatGPT-4, and lo cal LLMs, including CodeLlama-7B-Instruct, CodeLlama-13B Instruct, CodeLlama-34B-Instruct,</span><span class="c3">[</span><span class="c13">67</span><span class="c3">] </span><span class="c12">Llama-2-7b-chat, Llama-2- 13b-chat and Llama-2-70b-chat.</span><span class="c3">[</span><span class="c13">68</span><span class="c3">]</span><span class="c68">&nbsp;</span></p><p class="c228"><span class="c30">2.7. Security and Safety of AutoBA </span></p><p class="c179"><span class="c12">AutoBA incorporates a sandbox mode to establish a secure and isolated environment for conducting analyses. This mode encapsulates the analysis processes, effectively shielding the underlying system from potential threats. Meanwhile, AutoBA imposes restrictions on system commands throughout the execution phase, thereby reducing the risk of malicious com mands being executed within the environment. Additionally, AutoBA leverages Docker containerization, introducing an extra layer of security to further fortify the overall system integrity. Furthermore, Docker containerization simplifies the installa tion process, contributing to a reduction in learning costs for users. A workstation with 252 GB RAM, 112 CPU cores, and 1 Nvidia A100 GPU was adopted for all experiments. AutoBA was developed based on Python3.10 and CUDA12.0. CUDA is a par allel computing platform and programming model developed by NVIDIA. AutoBA utilizes LLMs as its core for generating analysis plans and code. These models benefit greatly from GPU acceler ation. CUDA allows for faster computation of model parameters, leading to quicker generation of results and more efficient han dling of complex natural language processing tasks. A detailed list of dependencies could be found in the code availability. The online version operates without the need for a GPU, while </span></p><p class="c129"><span class="c12">the offline version requires GPU support (7B: 12.55GB, 13B: 24GB, 34B: 63GB, 70B: 74GB). The sizes mentioned, such as 7B, 13B, 34B, and 70B, indicate the number of parameters in the models (B stands for billion). The corresponding GPU memory requirements in gigabytes (GB) are listed next to each model size. </span></p><p class="c189"><span class="c30">2.8. Statistical Analysis </span></p><p class="c255"><span class="c12">During the evaluation of AutoBA, all experimental data were ex amined at least three times on the same computational environ ment and data. Statistical analysis was performed using Python packages. </span></p><p class="c151"><span class="c16">3. Results </span></p><p class="c82"><span class="c30">3.1. AutoBA Proposes Detailed Analysis Plans for Tasks </span></p><p class="c63"><span class="c12">AutoBA offers a robust capability to generate a highly detailed and customized analysis plan, leveraging the user&rsquo;s input, which encompasses critical elements such as data paths, data descrip tions, and objective descriptions. </span></p><p class="c165"><span class="c12">As an example, in </span><span class="c30">Figure </span><span class="c18">3</span><span class="c12">, the user supplied four RNA-Seq samples: two from the LoGlu group (SRR1374921.fastq.gz and SRR1374922.fastq.gz, mouse pancreatic islets cultured at low ambient glucose) and two from the HiGlu group (SRR1374923.fastq.gz and SRR1374924.fastq.gz, mouse pan creatic islets cultured at high ambient glucose) from Ben ner et al.&rsquo;s paper.</span><span class="c3">[</span><span class="c13">69</span><span class="c3">] </span><span class="c12">Additionally, the user also provided the mouse reference genome (mm39.fa) and genome annotation (mm39.ncbiRefSeq.gtf ). The primary objective of this case was to identify differentially expressed genes between the two data groups. Using textual inputs only, AutoBA generated a detailed, step-by-step analysis plan during the planning phase, as outlined below: </span></p><p class="c80"><span class="c12">1) First, I will use the &ldquo;Trimmomatic&rdquo; software to trim the adapters from the single-end mouse rna-seq </span></p><p class="c201"><span class="c12">reads. The input files for this step will be &ldquo;./data/ </span></p><p class="c95"><span class="c12">SRR1374921.fastq.gz&rdquo;, &ldquo;./data/SRR1374922.fastq.gz&rdquo;, &ldquo;./data/SRR1374923.fastq.gz&rdquo;, </span></p><p class="c258"><span class="c12">&ldquo;./data/SRR1374924.fastq.gz&rdquo; and the adapter file &ldquo;./data/TruSeq3-SE.fa&rdquo;. </span></p><p class="c109"><span class="c12">2) Next, I will use the &ldquo;Hisat2&rdquo; software to align the trimmed reads to the mouse mm39 genome. The input files for this step will be the output files from the previous step and the genome file &ldquo;./data/mm39.fa&rdquo;. </span></p><p class="c278"><span class="c12">3) Then, I will use the &ldquo;Samtools&rdquo; software to convert the alignment output files from Hisat2 (in SAM format) to BAM format. </span></p><p class="c128"><span class="c12">4) After that, I will use the &ldquo;HTSeq&rdquo; software to count the num ber of reads mapped to each gene. The input files for this step will be the BAM files from the previous step and the annotation file &ldquo;./data/mm39.ncbiRefSeq.gtf&rdquo;. </span></p><p class="c171"><span class="c12">5) Finally, I will use the &ldquo;DESeq2&rdquo; software to identify the dif ferentially expressed genes. The input files for this step will be the count files from the previous step. </span></p><p class="c285"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (6 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c29"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span></p><p class="c29"><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c207"><span class="c2">Figure 3. </span><span class="c0">Example of applying AutoBA to find differentially expressed genes with RNA-Seq data. In this case, the user provided four RNA-seq samples, comprising two from the LoGlu group and two from the HiGlu group. The primary objective of this analysis was to identify differentially expressed genes between two groups. During the automated analysis prior to the alignment run by AutoBA, the ACR module identified an error where Hisat2 did not first index the genome. Consequently, the ACR module automatically corrected the code (highlighted in blue). This case demonstrates that AutoBA can be used for end-to-end bioinformatics analysis. Users only need to provide the simplest inputs, and AutoBA will deliver the final analysis results. </span></p><p class="c152"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (7 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c70"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c175"><span class="c12">Within each step of the plan outlined above, AutoBA provides precise instructions regarding the required software, including names like Trimmomatic, Hisat2, Samtools, HTSeq, and DE Seq2, along with clear sub-tasks for each analytical stage. This level of tailored planning ensures that the analysis process aligns precisely with the user&rsquo;s objectives, promoting both efficiency and accuracy in data processing and results generation. </span></p><p class="c97"><span class="c30">3.2. AutoBA Generates Precise Codes for Sub-Tasks </span></p><p class="c60"><span class="c12">During the code generation phase, AutoBA generates code in bash format for every sub-task of the plan established in the planning phase. These scripts encompass environment setup, software installation, and tailored code for software utilization. Parameters and data paths specific to the software are meticu lously incorporated. As exemplified in Figure </span><span class="c10">3</span><span class="c12">, the preliminary phase of the differentially expressed genes (DEG) analysis consti tutes the essential process of adapter trimming, an indispensable preprocessing step in the context of raw RNA-Seq data. Within this critical step, AutoBA automatically generated code, including activating the conda environment, installing software packages, and calling software to analyze data as shown below: </span></p><p class="c96"><span class="c12">source activate abc_runtime </span></p><p class="c48"><span class="c12">conda config &ndash;add channels conda-forge </span></p><p class="c48"><span class="c12">conda config &ndash;add channels bioconda </span></p><p class="c48"><span class="c12">conda install -y trimmomatic </span></p><p class="c102"><span class="c12">trimmomatic SE -phred33 ./data/SRR1374921.fastq.gz ./output/SRR1374921_trimmed.fastq.gz </span></p><p class="c77"><span class="c12">ILLUMINACLIP:./data/TruSeq3-SE.fa:2:30:10 </span></p><p class="c119"><span class="c12">trimmomatic SE -phred33 ./data/SRR1374922.fastq.gz ./output/SRR1374922_trimmed.fastq.gz </span></p><p class="c77"><span class="c12">ILLUMINACLIP:./data/TruSeq3-SE.fa:2:30:10 </span></p><p class="c119"><span class="c12">trimmomatic SE -phred33 ./data/SRR1374923.fastq.gz ./output/SRR1374923_trimmed.fastq.gz </span></p><p class="c77"><span class="c12">ILLUMINACLIP:./data/TruSeq3-SE.fa:2:30:10 </span></p><p class="c119"><span class="c12">trimmomatic SE -phred33 ./data/SRR1374924.fastq.gz ./output/SRR1374924_trimmed.fastq.gz </span></p><p class="c77"><span class="c12">ILLUMINACLIP:./data/TruSeq3-SE.fa:2:30:10 </span></p><p class="c43"><span class="c12">The generated code further underwent a meticulous and thor ough validation process, which included a manual verification and execution performed by experienced and proficient bioinfor maticians, as meticulously documented in </span><span class="c30">Table </span><span class="c18">3 </span><span class="c12">and Supple mentary information. This critical validation step ensures the ac curacy and reliability of the code, reaffirming the robustness of AutoBA. </span></p><p class="c97"><span class="c30">3.3. AutoBA Adeptly Manages Similar Tasks with Robustness </span></p><p class="c272"><span class="c12">In practical bioinformatics applications, even when researchers are working with similar data types, such as RNA-Seq, it is note worthy that analyses often manifest variations stemming from diverse sources. These variations are primarily attributed to dis parities in the characteristics of input data and the distinct objec tives pursued in the analytical process. </span></p><p class="c33"><span class="c12">As exemplified in Case 10.1 (find differentially expressed genes), Case 10.2 (identify the top five down-regulated genes in HiGlu group), and Case 10.3 (predict fusion genes), when per forming RNA-Seq analysis, users may have distinct final goals, necessitating adjustments in software and parameter selection during the actual execution. In comparison to case 10.1, AutoBA introduces an additional step in case 10.2, tailored for screening the top five differentially expressed genes to fulfill the user&rsquo;s spe cific requirements as shown in the code below: </span></p><p class="c40"><span class="c12">Rscript -e &lsquo;&lsquo;library(&rsquo;pheatmap&rsquo;); library(&rsquo;DESeq2&rsquo;); </span></p><p class="c20"><span class="c12">res</span><span class="c53">&lt;</span><span class="c12">- read.csv(&rsquo;./examples/output/differential_expression_ results.csv&rsquo;, row.names=1); res_ordered</span><span class="c53">&lt;</span><span class="c12">- </span></p><p class="c25"><span class="c12">res[order(res$log2FoldChange),]; top5_downregulated</span><span class="c53">&lt;</span><span class="c12">- head(res_ordered, 5); </span></p><p class="c212"><span class="c30">3.4. AutoBA Adjusts Analysis Based on Task and Input Data Variations </span></p><p class="c183"><span class="c12">Alignment is an essential step for bioinformatic analysis, for which multiple tools have been developed for distinct tasks. For instance, tools including STAR</span><span class="c3">[</span><span class="c13">70</span><span class="c3">] </span><span class="c12">and HISAT2</span><span class="c3">[</span><span class="c13">71</span><span class="c3">] </span><span class="c12">designed for RNA-seq data analysis are splicing aware, which is efficient in identifying junction reads that map to two distal positions in the reference genome. Besides, long-read sequencing data from Pacific Bioscience (PacBio) and Oxford Nanopore Technology (ONT) also require specialized tools for the alignment, for which Minimap2</span><span class="c3">[</span><span class="c13">72</span><span class="c3">] </span><span class="c12">is the most widely used method. Moreover, each read from single-cell sequencing data contains barcodes for UMI and cell labels, which needs to be integrated with the alignment. CellRanger is a popular software with this capacity. Therefore, bioinformatic analysis should use appropriate tools for the align ment based on the types of tasks. Interestingly, we found that AutoBA has learned this knowledge and can correctly employ the tool for the alignment (</span><span class="c30">Figure </span><span class="c18">4</span><span class="c10">a</span><span class="c12">). </span></p><p class="c115"><span class="c12">For many bioinformatic analyses, multiple tools are available but require different conditions of inputs. For instance, to iden tify structural variations from tumor WGS/WES data, the method &ldquo;manta&rdquo;</span><span class="c3">[</span><span class="c13">73</span><span class="c3">] </span><span class="c12">can handle the analysis against the matched normal. On the other hand, tools like &ldquo;Pindel&rdquo;</span><span class="c3">[</span><span class="c13">74</span><span class="c3">] </span><span class="c12">that relies on the de tection of breakpoints with the reference genome, only conduct analysis on the tumor samples. We found that AutoBA can auto matically select &ldquo;manta&rdquo; when the matched normal samples were provided and correctly utilized the parameters &ldquo;&ndash;normalBam&rdquo; and &ldquo;&ndash;tumorBam&rdquo;. However, if only the tumor samples were pro vided in the input data, AutoBA will select &ldquo;Pindel&rdquo; for the anal ysis (Figure </span><span class="c10">4b</span><span class="c12">). These results suggest that AutoBA learned the requirements of different bioinformatic tools and is capable of selecting appropriate tools based on different conditions of the input data. </span></p><p class="c92"><span class="c12">manta &ndash;normalBam ./output/SRR23015874.recalibrated.bam &ndash;tumorBam ./output/SRR23015876.recalibrated.bam &ndash;referenceFasta ./data/hg38.fa &ndash;runDir ./out </span></p><p class="c77"><span class="c12">put/manta_SRR23015874 </span></p><p class="c21"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (8 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c29"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span></p><p class="c29"><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c225"><span class="c2">Table 3. </span><span class="c0">Summary of AutoBA (w/o ACR) generated results evaluated by bioinformatics experts. The table presents an assessment conducted by bioinfor matics experts on the analysis plan proposed by AutoBA, along with the generated codes and the code execution. If the evaluation passes, it is displayed as success, while instances of failure are accompanied by detailed explanations of the specific reasons for the failure. Additionally, we provide a summary of the software tools automatically chosen by AutoBA for each case, as well as the total time taken to generate the corresponding code. </span></p><p class="c145"><span class="c5">Case ID Propose Plans Generate Codes Execute Codes Tools Used Time Cost (without Executing Codes) in </span></p><p class="c29"><span class="c5">Minutes </span></p><p class="c262"><span class="c5">1.1 Success Success Success FastQC, Trimmomatic</span><span class="c15">[</span><span class="c8">76</span><span class="c15">]</span><span class="c5">, SPAdes</span><span class="c15">[</span><span class="c8">77</span><span class="c15">]</span><span class="c5">, QUAST</span><span class="c15">[</span><span class="c8">78</span><span class="c142">] </span><span class="c5">3 2.1 Success Success Success FastQC, Trimmomatic, BWA</span><span class="c15">[</span><span class="c8">79</span><span class="c15">]</span><span class="c5">, Samtools</span><span class="c15">[</span><span class="c8">80</span><span class="c15">]</span><span class="c5">, GATK</span><span class="c15">[</span><span class="c8">81</span><span class="c15">] </span><span class="c5">8 </span></p><p class="c38 c248"><span class="c5">2.2 Success Success Success FastQC, Trimmomatic, BWA, Samtools, GATK, </span></p><p class="c79"><span class="c5">8 </span></p><p class="c126"><span class="c5">ensembl-vep</span><span class="c15">[</span><span class="c8">82</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c28 c38"><span class="c5">2.3 Success Success Success FastQC, Trimmomatic, BWA, Samtools, GATK, manta</span><span class="c15">[</span><span class="c8">73</span><span class="c15">] </span><span class="c5">18 </span></p><p class="c56"><span class="c5">2.4 Success Success Failed: pindel requires configuration file </span></p><p class="c28 c280"><span class="c5">FastQC, Trimmomatic, BWA, Samtools, pindel</span><span class="c15">[</span><span class="c8">74</span><span class="c15">]</span><span class="c5">, </span></p><p class="c66"><span class="c5">6 </span></p><p class="c28 c224"><span class="c5">SnpEff</span><span class="c15">[</span><span class="c8">83</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c38 c185"><span class="c5">3.1 Success Success Success FastQC, Trim Galore, Bowtie 2</span><span class="c15">[</span><span class="c8">84</span><span class="c15">]</span><span class="c5">, Samtools, MACS2</span><span class="c15">[</span><span class="c8">85</span><span class="c15">]</span><span class="c5">, </span></p><p class="c79"><span class="c5">6 </span></p><p class="c71"><span class="c5">BEDTools, IGV </span></p><p class="c38 c295"><span class="c5">3.2 Success Success Success FastQC, Trim Galore, Bowtie2, MACS2, HOMER, </span></p><p class="c79"><span class="c5">4 </span></p><p class="c49"><span class="c5">MEME</span><span class="c15">[</span><span class="c8">86</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c81"><span class="c5">3.3 Failed: DESeq2 is not suitable for </span></p><p class="c73"><span class="c5">peaks identified </span></p><p class="c28"><span class="c5">by MACS2 </span></p><p class="c28"><span class="c5">&ndash; &ndash; FastQC, BWA, MACS, BEDTools</span><span class="c15">[</span><span class="c8">87</span><span class="c15">]</span><span class="c5">, DESeq2</span><span class="c15">[</span><span class="c8">88</span><span class="c15">]</span><span class="c5">, </span></p><p class="c28"><span class="c5">6 </span></p><p class="c28"><span class="c5">g:Profiler</span><span class="c15">[</span><span class="c8">89</span><span class="c15">]</span><span class="c5">, R</span><span class="c15">[</span><span class="c8">90</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c195"><span class="c5">4.1 Success Success Success Trim Galore, Bismark</span><span class="c15">[</span><span class="c8">91</span><span class="c15">]</span><span class="c5">, IGV</span><span class="c15">[</span><span class="c8">92</span><span class="c15">] </span><span class="c5">9 </span></p><p class="c81 c219"><span class="c5">5.1 Success Success Failed: (wrongly used BEDTools) </span></p><p class="c251"><span class="c5">Trim Galore, BWA, Samtools, MACS2, BEDTools 8 </span></p><p class="c159"><span class="c5">6.1 Success Success Success FastQC, Cutadapt, BWA, MACS2, IGV, GREAT</span><span class="c15">[</span><span class="c8">93</span><span class="c15">] </span><span class="c5">5 7.1 Success Success Success FastQC, BEDTools, Samtools, Bowtie 2, R 6 </span></p><p class="c154"><span class="c5">8.1 Success Success Failed: racon medaka wrongly used the </span></p><p class="c125"><span class="c5">parameters </span></p><p class="c28 c293"><span class="c5">canu</span><span class="c15">[</span><span class="c8">94</span><span class="c15">]</span><span class="c5">, Minimap2</span><span class="c15">[</span><span class="c8">72</span><span class="c15">]</span><span class="c5">, Racon</span><span class="c15">[</span><span class="c8">95</span><span class="c15">]</span><span class="c5">, Flye</span><span class="c15">[</span><span class="c8">96</span><span class="c15">]</span><span class="c5">, Medaka, </span></p><p class="c66"><span class="c5">7 </span></p><p class="c28 c203"><span class="c5">Bandage</span><span class="c15">[</span><span class="c8">97</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c138"><span class="c5">8.2 Failed: cannot find a correct </span></p><p class="c73"><span class="c5">pipeline </span></p><p class="c28"><span class="c5">&ndash; &ndash; Minimap2, Samtools, trf</span><span class="c15">[</span><span class="c8">98</span><span class="c15">] </span><span class="c5">7 </span></p><p class="c197"><span class="c5">9.1 Success Failed: install the wrong tool, </span></p><p class="c73"><span class="c5">pb-falcon rather </span></p><p class="c28"><span class="c5">than falcon </span></p><p class="c28"><span class="c5">&ndash; Canu, FALCON</span><span class="c15">[</span><span class="c8">99</span><span class="c15">]</span><span class="c5">, Quiver, MUMmer</span><span class="c15">[</span><span class="c8">100</span><span class="c15">] </span><span class="c5">7 </span></p><p class="c233 c292"><span class="c5">10.1 Success Success Success FASTQC, Trimmomatic, HISAT2</span><span class="c15">[</span><span class="c8">71</span><span class="c15">]</span><span class="c5">, htseq</span><span class="c15">[</span><span class="c8">101</span><span class="c15">]</span><span class="c5">, DESeq2 5 10.2 Success Success Success FASTQC, Trimmomatic, HISAT2, htseq, DESeq2, gprofileR 5 10.3 Success Success Success gunzip, HISAT2, fusioncatcher</span><span class="c15">[</span><span class="c8">102</span><span class="c15">]</span><span class="c5">, gffcompare</span><span class="c15">[</span><span class="c8">103</span><span class="c15">] </span><span class="c5">6 10.4 Success Success Success Trim Galore, HISAT2, Samtools, StringTie 5 </span></p><p class="c167"><span class="c5">10.5 Success Success Success Trimmomatic, HISAT2, Samtools, StringTie, </span></p><p class="c79"><span class="c5">6 </span></p><p class="c140"><span class="c5">featureCounts</span><span class="c15">[</span><span class="c8">36</span><span class="c15">]</span><span class="c5">, rMATs</span><span class="c15">[</span><span class="c8">38</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c81"><span class="c5">10.6 Success Failed: DaPars (not available in conda) </span></p><p class="c28"><span class="c5">&ndash; Trim Galore, HISAT2, StringTie, DaPars</span><span class="c15">[</span><span class="c8">104</span><span class="c15">] </span><span class="c5">7 </span></p><p class="c138"><span class="c5">10.7 Failed: cannot find a correct </span></p><p class="c73"><span class="c5">pipeline </span></p><p class="c28"><span class="c5">&ndash; &ndash; FastQC, Trimmomatic, HISAT2, Samtools, StringTie, </span></p><p class="c28"><span class="c5">7 </span></p><p class="c28"><span class="c5">ballgowan</span><span class="c15">[</span><span class="c8">105</span><span class="c15">]</span><span class="c5">, GATK </span></p><p class="c110"><span class="c5">10.8 Success Failed: CIRI2 (not available in conda) </span></p><p class="c28"><span class="c5">&ndash; Trim Galore, HISAT2, CIRI2</span><span class="c15">[</span><span class="c8">106</span><span class="c15">]</span><span class="c5">, CIRIQuant</span><span class="c15">[</span><span class="c8">107</span><span class="c15">] </span><span class="c5">5 </span></p><p class="c231"><span class="c5">11 </span></p><p class="c104"><span class="c5">11.1 Success Success Success Fastqc, Cutadapt, Bowtie, Samtools, subread/featureCounts, DESeq2, edgeR</span><span class="c15">[</span><span class="c8">108</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c22"><span class="c5">11.2 Success Success Failed: conda of miRDeep2 is </span></p><p class="c73"><span class="c5">problematic </span></p><p class="c139"><span class="c5">Fastqc, Cutadapt, Bowtie, Samtools, featureCounts, miRDeep2, DESeq2, edgeR </span></p><p class="c28"><span class="c5">11 </span></p><p class="c112"><span class="c5">(</span><span class="c17">Continued</span><span class="c5">) </span></p><p class="c184"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (9 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c29"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span></p><p class="c29"><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c270"><span class="c2">Table 3. </span><span class="c0">(Continued) </span></p><p class="c243"><span class="c5">Case ID Propose Plans Generate Codes Execute Codes Tools Used Time Cost (without Executing Codes) in </span></p><p class="c29"><span class="c5">Minutes </span></p><p class="c193"><span class="c5">12.1 Success Success Success Fastqc, Trimmomatic, HISAT2, HTSeq/htseq-count, </span></p><p class="c169"><span class="c5">6 </span></p><p class="c122"><span class="c5">CAGEr</span><span class="c15">[</span><span class="c8">109</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c244"><span class="c5">13.1 Failed: cannot find a correct </span></p><p class="c73"><span class="c5">pipeline </span></p><p class="c28"><span class="c5">&ndash; &ndash; Trim Galore, HISAT2, StringTie, DaPars 5 </span></p><p class="c105"><span class="c5">14.1 Success Success Failed: prepDE.py no need to run with </span></p><p class="c176"><span class="c5">&rsquo;python prepDE.py&rsquo; </span></p><p class="c44"><span class="c5">Minimap2, Samtools, StringTie, DESeq2 9 </span></p><p class="c130"><span class="c5">15.1 Success Success Success Minimap2, Samtools, StringTie, cufflinks</span><span class="c15">[</span><span class="c8">110</span><span class="c15">] </span><span class="c5">5 </span></p><p class="c81 c164"><span class="c5">16.1 Success Success Failed: conda of Piranha is </span></p><p class="c176 c276"><span class="c5">problematic </span></p><p class="c44"><span class="c5">FastQC, Cutadapt, Bowtie2, Samtools, BEDTools, Piranha 6 </span></p><p class="c46"><span class="c5">16.2 Success Success Success FastQC, Trim Galore, HISAT2, htseq, DESeq2 4 </span></p><p class="c116"><span class="c5">17.1 Success Success Failed: not regular conda of ribotaper </span></p><p class="c28 c78"><span class="c5">FastQC, Trim Galore, HISAT2, Samtools, StringTie, </span></p><p class="c44"><span class="c5">7 </span></p><p class="c23"><span class="c5">RiboTaper</span><span class="c15">[</span><span class="c8">111</span><span class="c15">]</span><span class="c19">&nbsp;</span></p><p class="c235 c279"><span class="c5">18.1 Success Success Success Cell Ranger, Seurat</span><span class="c15">[</span><span class="c8">112</span><span class="c15">] </span><span class="c5">5 18.2 Success Success Success Scanpy</span><span class="c15">[</span><span class="c8">113</span><span class="c15">] </span><span class="c5">8 18.3 Success Success Success Scanpy 6 18.4 Success Success Success Scanpy 5 19.1 Success Success Success Squidpy</span><span class="c15">[</span><span class="c8">114</span><span class="c15">]</span><span class="c5">, AnnData 5 19.2 Success Success Success AnnData, Scanpy, Tangram</span><span class="c15">[</span><span class="c8">115</span><span class="c15">] </span><span class="c5">3 20.1 Success Success Success proteowizard</span><span class="c15">[</span><span class="c8">116</span><span class="c15">]</span><span class="c5">, OpenMS</span><span class="c15">[</span><span class="c8">117</span><span class="c15">] </span><span class="c5">15 21.1 Success Success Success pymzml</span><span class="c15">[</span><span class="c8">118</span><span class="c15">]</span><span class="c5">, pandas, numpy, scipy 13 #Success 36 33 26 &ndash; &ndash; </span></p><p class="c101"><span class="c30">3.5. Apply AutoBA to a Variety of Conventional Multi-Omic Analysis Scenarios </span></p><p class="c239 c241"><span class="c12">To evaluate the robustness of AutoBA, we conducted assessments involving a total of 40 cases spanning four distinct types of omics data: genomics, transcriptomics, proteomics, and metabolomics as shown in Table </span><span class="c10">2 </span><span class="c12">and Supporting Information. </span></p><p class="c268"><span class="c12">All cases underwent an independent analysis process con ducted by AutoBA and were subsequently subjected to validation by experienced bioinformatics experts. The collective results un derscore the versatility and robustness of AutoBA across a spec trum of multi-omics analysis procedures in the field of bioinfor matics as shown in Table </span><span class="c10">3</span><span class="c12">. AutoBA demonstrates its capability to autonomously devise novel analysis processes based on vary ing input data, showcasing its adaptability to diverse input data and analysis objectives with a success rate of 90% (36 out of 40) for proposing plans, 82.5% (33 out of 40) for generating codes to obtain and install appropriate tools, and 65% (26 out of 40) for automated end-to-end analysis. With the incorporation of the ACR module, AutoBA demonstrates enhanced robustness, with the same success rate of 90% (36 out of 40) for proposing plans, but a higher success rate of 87.5% (35 out of 40) for generating codes to obtain and install appropriate tools, and 87.5% (35 out of 40) for automated end-to-end analysis. Compared to the online version, the local version showed a slight decline in performance as shown in Figure </span><span class="c10">4d</span><span class="c12">. </span></p><p class="c131"><span class="c30">3.6. AutoBA Reduces Human Intervention and Increases Robustness Compared to Other Methods </span></p><p class="c99"><span class="c12">As shown in Figure </span><span class="c10">4c</span><span class="c12">, we conducted a conceptual comparison between AutoBA and alternative methods in terms of human intervention. In utilizing conventional bioinformatics tools and web servers, users are required to prepare input data and com prehend detailed analysis plans prior to execution. Throughout the execution phase, users must configure the environment, install essential dependencies, write code, and proceed with step-by-step debugging. In contrast, ChatGPT and other open source LLMs assist users in proposing step-by-step plans and generating code, thus mitigating human intervention. Never theless, users still need to manually configure the environment, execute code, and perform debugging. AutoGPT, functioning as an AI agent, aids users in executing generated code to further minimize human intervention. However, within the context of bioinformatics data analysis, AutoGPT encounters challenges in setting up the environment and debugging for users. Conversely, AutoBA significantly reduces human intervention, necessitating only the preparation of input data. </span></p><p class="c223"><span class="c12">To show the robustness of AutoBA, we further conducted a comprehensive comparison of eight methods, including 1) AutoBA (w/o ACR, online with ChatGPT-4), 2) AutoBA (with ACR, online with ChatGPT-4), 3) AutoBA (w/o ACR, offline with CodeLlama-34B-Instruct), 4) AutoBA (with ACR, offline </span></p><p class="c180"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (10 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c29"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 435.23px; height: 201.65px;"><img alt="" src="images/image4.png" style="width: 435.23px; height: 201.65px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span></p><p class="c29"><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c107"><span class="c2">Figure 4. </span><span class="c0">Results of AutoBA and the comparison with other methods. a) Heatmap illustrating options of utilizing different alignment tools for multiple tasks planned by AutoBA. b) AutoBA utilizes the tools for identifying structure variations in tumor samples with or without the matched normal samples. The highlight shows the difference between Goal 1 and Goal 2. c) Conceptual comparison of AutoBA with other methods in terms of human intervention. Orange indicates the need for human intervention, while green signifies an absence of human intervention (fully automated process). d) Evaluation of results generated by various methods by manually checking and executing codes and comparing them to standard analysis pipelines. Orange indicates a failure, and blue indicates a success. </span></p><p class="c160"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (11 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c70"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c61"><span class="c12">with CodeLlama-34B-Instruct), 5) AutoGPT, 6) ChatGPT-3.5, 7) ChatGPT-4 and 8) CodeLlama-34B-Instruct, across all 40 cases, as illustrated in Figure </span><span class="c10">4d</span><span class="c12">. AutoBA showed better performance in comparison to AutoGPT (90% for proposing plans, 25% for gen erating codes to obtain and install appropriate tools, and 0% for automated end-to-end analysis), ChatGPT-3.5 (92.5% for propos ing plans, 30% for generating codes to obtain and install ap propriate tools, and 2.5% for automated end-to-end analysis), ChatGPT-4 (92.5% for proposing plans, 37.5% for generating codes to obtain and install appropriate tools, and 7.5% for au tomated end-to-end analysis), and CodeLlama-34B-Instruct (80% for proposing plans, 7.5% for generating codes to obtain and in stall appropriate tools, and 2.5% for automated end-to-end anal ysis). </span></p><p class="c239 c275"><span class="c16">4. Discussion </span></p><p class="c90"><span class="c12">To our knowledge, AutoBA is the first autonomous AI agent tai lored explicitly for conventional multi-omic analyses for omics data. AutoBA streamlines the analytical process, requiring min imal user input while providing detailed step-by-step plans for various bioinformatics tasks (Video </span><span class="c10">S1</span><span class="c12">, Supporting Information). The results of our investigation reveal that AutoBA excels in ac curately handling a diverse array of omics analysis tasks, such as RNA-seq, scRNA-seq, ChIP-seq, spatial transcriptomics, and so on. One of the key strengths of AutoBA is its adaptability to variations in analysis objectives. As demonstrated in the cases presented, even with similar data types, such as RNA-Seq, users often have distinct goals, necessitating modifications in software and parameter selection during execution. AutoBA effectively ac commodates these variations, allowing users to tailor their anal yses to specific research needs without compromising accuracy. Furthermore, AutoBA&rsquo;s versatility is highlighted by its ability to self-design new analysis processes based on differing input data. This autonomous adaptability makes AutoBA a valuable tool for bioinformaticians working on novel or unconventional research questions, as it can adjust its approach to the unique characteris tics of the data. </span></p><p class="c36"><span class="c12">Online bioinformatics analysis platforms are currently in vogue, but they often necessitate the uploading of either raw data or pre-processed statistics by users, which could potentially give rise to privacy concerns and data leakage risks. In contrast, Au toBA addresses these privacy issues by offering both online ver sion and local version. When utilizing the online version of Au toBA with ChatGPT, data uploads are unnecessary, requiring only descriptive information in natural language as specified in our prompt design. This information is limited in terms of private details. In comparison, the local version of AutoBA provides the highest level of privacy protection, as it operates on local back ends and eliminates the need to share any information with third parties. Moreover, AutoBA showcases its adaptability in sync with emerging bioinformatics tools, with LLM seamlessly incorporat ing these latest tools into the database. Furthermore, AutoBA is inclined toward selecting the most popular analytical frameworks or widely applicable tools in the planning phase, underscoring its robustness. Another distinguishing feature is AutoBA&rsquo;s transpar ent and interpretable execution process. This transparency allows professional bioinformaticians to easily modify and customize </span></p><p class="c229"><span class="c12">AutoBA&rsquo;s outputs, leveraging AutoBA to expedite the data analy sis process. </span></p><p class="c132"><span class="c12">AutoBA is also a future-proof AI agent designed for bioin formatics analysis, leveraging LLMs as its core. This design al lows AutoBA to integrate with any existing LLM, whether online (e.g., ChatGPT, GPT-4, GPT-4o) or offline (e.g., LLaMA, CodeL LaMA, and DeepSeek). The LLM used in AutoBA is fully substi tutable, enabling it to benefit from the continual advancements in LLM technology. As new state-of-the-art LLMs are developed, AutoBA can incorporate them to enhance its performance in au tomatic bioinformatics analysis. Still, AutoBA&rsquo;s limitation in tool selection does exist. Current LLMs are trained on internet data, meaning that widely used methods in bioinformatics are typi cally well-trained, while methods from specific papers may be underrepresented or not trained at all. As a result, the best out comes are achieved when using tools that have been extensively trained, which can lead to potential biases in tool selection. To address this, training a specialized LLM for bioinformatics that thoroughly covers all tools and methods in the field could be a solution in the future. </span></p><p class="c205"><span class="c12">Given that classical bioinformatic analysis encompasses a far broader spectrum of tasks and challenges than the 40 cases stud ied in this work (Tables </span><span class="c10">2 </span><span class="c12">and </span><span class="c10">3</span><span class="c12">), it is essential to conduct more real-world applications by our potential users to further com prehensively validate the robustness of AutoBA. We found that a large proportion (36%, 5 out of 14) of failed cases in execut ing code is due to the tools in conda being problematic, not in a regular form (end with .sh, .pl et al), or requiring an edited config file, suggesting a demand for more standard bioinformat ics tools. Furthermore, taking into account the timeliness of the training data used for large language models, it&rsquo;s important to note that some of the most recently proposed methods in bioin formatics may still pose challenges in automatically generating code by AutoBA. Therefore, a future endeavor to train an up-to date large language model explicitly tailored for bioinformatics can significantly enhance AutoBA&rsquo;s ability to maintain up-to-date code generation capabilities. Nevertheless, AutoBA represents a significant advancement in the field of bioinformatics, offering a user-friendly, efficient, and adaptable solution for a wide range of omics analysis tasks. Its capacity to handle diverse data types and analysis goals, coupled with its robustness and adaptabil ity, positions AutoBA as a valuable asset in the pursuit of accel erating bioinformatics research. We anticipate that AutoBA will find extensive utility in the scientific community, supporting re searchers in their quest to extract meaningful insights from com plex biological data. </span></p><p class="c155"><span class="c16">Supporting Information </span></p><p class="c194"><span class="c0">Supporting Information is available from the Wiley Online Library or from the author. </span></p><p class="c72"><span class="c16">Acknowledgements </span></p><p class="c208"><span class="c0">J.Z., B.Z., X.C., H.L., X.X., S.C., W.H., C.X., and X.G. were supported in part by grants from the Office of Research Administration (ORA) at King Abdul lah University of Science and Technology (KAUST) under award number FCC/1/1976-44-01, FCC/1/1976-45-01, REI/1/5202-01-01, REI/1/5234-01- 01, REI/1/4940-01-01, RGC/3/4816-01-01, REI/1/0018-01-01, REI/1/5414- 01-01, REI/1/5289-01-01, and REI/1/5404-01-01. </span></p><p class="c91"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (12 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c70"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 118.27px; height: 28.83px;"><img alt="" src="images/image1.png" style="width: 118.27px; height: 28.83px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c26"><span class="c16">Conflict of Interest </span></p><p class="c134 c191"><span class="c0">The authors declare no conflict of interest. </span></p><p class="c253"><span class="c16">Author Contributions </span></p><p class="c134 c269"><span class="c0">J.Z. and B.Z. contributed equally to this work. J.Z. and X.G performed con ceptualization. J.Z., B.Z., and X.G. performed design. J.Z. performed code Implementation. J.Z., B.Z., X.C., H.L., C.X., and W.H. performed applica tion. J.Z. and B.Z. wrote the original draft of the manuscript. J.Z., B.Z., X.X., S.C., X.G., L.L., and G.L. performed critical revision of the manuscript for important intellectual content. J.Z. and X.G. performed supervision. X.G. performed funding acquisition. </span></p><p class="c54"><span class="c16">Code Availability Statement </span></p><p class="c233 c134 c281"><span class="c0">The AutoBA software is publicly available at </span><span class="c31">https://github.com/ JoshuaChou2018/AutoBA</span><span class="c0">. The Docker version of AutoBA is available at </span><span class="c31">https://hub.docker.com/r/joshuachou666/autoba</span><span class="c0">. </span></p><p class="c188"><span class="c16">Data Availability Statement </span></p><p class="c134 c211"><span class="c0">The RNA-seq dataset could be downloaded from Sequence Read Archive (SRA) with IDs: SRR1374921, SRR1374922, SRR1374923, and SRR1374924. The dataset for case 1.3 could be downloaded from </span><span class="c31">https: //github.com/STAR-Fusion/STAR-Fusion-Tutorial/wiki</span><span class="c0">. The scRNA-seq dataset could be downloaded from </span><span class="c31">http://cf.10xgenomics.com/samples/ cell-exp/1.1.0/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz</span><span class="c0">. The ChIP-seq dataset could be downloaded with IDs: SRR620204, SRR620205, SRR620206, and SRR620208. The Spatial Transcriptomics dataset could be downloaded from </span><span class="c31">https://doi.org/10.5281/zenodo.6334774</span><span class="c0">. The CAGE seq dataset could be downloaded from SRA with IDs: SRR11351697, SRR11351698, SRR11351700, and SRR11351701. The 3&rsquo;end-seq dataset could be downloaded from SRA with IDs: SRR17422754, SRR17422755, SRR17422756, and SRR17422757. The CLIP-seq dataset could be downloaded from ENCODE (</span><span class="c31">https://www.encodeproject.org</span><span class="c0">) with IDs: ENCLB742AYH and ENCLB770EDJ. The Ribo-seq data could be downloaded from SRA with IDs: RR12354645 and RR12354646. The raw single-cell RNA sequencing data could be downloaded from 10X genomics. The PacBio long-read sequencing data could be downloaded from SRA with IDs: SRR19552218 and SRR19785215. The small RNA-seq data could be downloaded from the previous study.</span><span class="c3">[</span><span class="c13">75</span><span class="c3">]</span><span class="c68">&nbsp;</span></p><p class="c188"><span class="c16">Keywords </span></p><p class="c191 c235"><span class="c0">agent, bioinformatics, large language model, omics analysis </span></p><p class="c174"><span class="c0">Received: June 25, 2024 </span></p><p class="c135"><span class="c0">Revised: August 11, 2024 </span></p><p class="c135"><span class="c0">Published online: October 3, 2024 </span></p><p class="c118"><span class="c0">[1] N. M. Luscombe, D. Greenbaum, M. Gerstein, </span><span class="c1">Methods of informa tion in medicine </span><span class="c2">2001</span><span class="c0">, </span><span class="c1">40</span><span class="c0">, 346. </span></p><p class="c9"><span class="c0">[2] J. Gauthier, A. T. Vincent, S. J. Charette, N. Derome, </span><span class="c1">Briefings in bioin formatics </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">20</span><span class="c0">, 1981. </span></p><p class="c57"><span class="c0">[3] A. D. Baxevanis, G. D. Bader, D. S. Wishart, </span><span class="c1">Bioinformatics</span><span class="c0">, John Wi ley &amp; Sons, NJ, USA </span><span class="c2">2020</span><span class="c0">. </span></p><p class="c150"><span class="c0">[4] P. Munk, C. Brinch, F. D. M&oslash;ller, T. N. Petersen, R. S. Hendriksen, A. M. Seyfarth, J. S. Kjeldgaard, C. A. Svendsen, B. van Bunnik, F. Berglund, </span><span class="c1">Nat. Commun. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">13</span><span class="c0">, 7251. </span></p><p class="c161"><span class="c0">[5] E. H. Lips, T. Kumar, A. Megalios, L. L. Visser, M. Sheinman, A. Fortunato, V. Shah, M. Hoogstraat, E. Sei, D. Mallo, </span><span class="c1">Nat. Genet. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">54</span><span class="c0">, 850. </span></p><p class="c157"><span class="c0">[6] D. T. Jones, J. M. Thornton, </span><span class="c1">Nat. Methods </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">19</span><span class="c0">, 15. [7] M. L. Hekkelman, I. de Vries, R. P. Joosten, A. Perrakis, </span><span class="c1">Nat. Methods </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">20</span><span class="c0">, 205. </span></p><p class="c246"><span class="c0">[8] N. Sapoval, A. Aghazadeh, M. G. Nute, D. A. Antunes, A. Balaji, R. Baraniuk, C. Barberan, R. Dannenfelser, C. Dun, M. Edrisi, </span><span class="c1">Nat. Commun. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">13</span><span class="c0">, 1728. </span></p><p class="c264"><span class="c0">[9] T. Gupta, M. Zaki, N. A. Krishnan, Mausam, </span><span class="c1">npj Comput. Mater. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">8</span><span class="c0">, 102. </span></p><p class="c252 c260"><span class="c0">[10] Z. Zeng, Y. Yao, Z. Liu, M. Sun, </span><span class="c1">Nat. Commun. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">13</span><span class="c0">, 862. [11] N. De Maio, P. Kalaghatgi, Y. Turakhia, R. Corbett-Detig, B. Q. Minh, N. Goldman, </span><span class="c1">Nat. Genet. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">55</span><span class="c0">, 746. </span></p><p class="c252 c289"><span class="c0">[12] A. S. Chanderbali, L. Jin, Q. Xu, Y. Zhang, J. Zhang, S. Jian, E. Carroll, D. Sankoff, V. A. Albert, D. G. Howarth, </span><span class="c1">Nat. Commun. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">13</span><span class="c0">, 643. [13] J. Rhodes, A. Abdolrasouli, K. Dunne, T. R. Sewell, Y. Zhang, E. Ballard, A. P. Brackin, N. van Rhijn, H. Chown, A. Tsitsopoulou, </span><span class="c1">Nat. Microbiol. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">7</span><span class="c0">, 663. </span></p><p class="c242"><span class="c0">[14] A. Heinken, J. Hertel, G. Acharya, D. A. Ravcheev, M. Nyga, O. E. Okpala, M. Hogan, S. Magn&uacute;sd&oacute;ttir, F. Martinelli, B. Nap, </span><span class="c1">Nat. Biotechnol. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">41</span><span class="c0">, 1320. </span></p><p class="c27"><span class="c0">[15] F. Hemmerling, J. Piel, </span><span class="c1">Nat. Rev. Drug Discovery </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">21</span><span class="c0">, 359. [16] J. Zhou, B. Zhang, H. Li, L. Zhou, Z. Li, Y. Long, W. Han, M. Wang, H. Cui, J. Li, </span><span class="c1">Genomics, Proteomics and Bioinformatics </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">20</span><span class="c0">, 959. [17] H. Li, H. Li, J. Zhou, X. Gao, </span><span class="c1">Bioinformatics </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">38</span><span class="c0">, 4878. [18] T. Zhang, L. Li, H. Sun, D. Xu, G. Wang, </span><span class="c1">Briefings in Bioinformatics </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">24</span><span class="c0">, bbad316. </span></p><p class="c273"><span class="c0">[19] Z. Li, E. Gao, J. Zhou, W. Han, X. Xu, X. Gao, </span><span class="c1">Cell Reports Methods </span><span class="c2">2023</span><span class="c0">, 3. </span></p><p class="c263"><span class="c0">[20] Y. Long, B. Zhang, S. Tian, J. J. Chan, J. Zhou, Z. Li, Y. Li, Z. An, X. Liao, Y. Wang, </span><span class="c1">Genome Res. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">33</span><span class="c0">, 644. </span></p><p class="c149"><span class="c0">[21] A. F. Bardet, Q. He, J. Zeitlinger, A. Stark, </span><span class="c1">Nat. Protoc. </span><span class="c2">2012</span><span class="c0">, </span><span class="c1">7</span><span class="c0">, 45. [22] B. Vieth, S. Parekh, C. Ziegenhain, W. Enard, I. Hellmann, </span><span class="c1">Nat. Com mun. </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">10</span><span class="c0">, 4667. </span></p><p class="c210"><span class="c0">[23] M. D. Luecken, F. J. Theis, </span><span class="c1">Molecular systems biology </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">15</span><span class="c0">, e8746. [24] F. C. Grandi, H. Modi, L. Kampman, M. R. Corces, </span><span class="c1">Nat. Protoc. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">17</span><span class="c0">, 1518. </span></p><p class="c186"><span class="c0">[25] P. C. Ng, E. F. Kirkness, </span><span class="c1">Genetic variation: Methods and protocols </span><span class="c2">2010</span><span class="c0">, </span><span class="c1">628</span><span class="c0">, 215. </span></p><p class="c247"><span class="c0">[26] Z. Wang, M. Gerstein, M. Snyder, </span><span class="c1">Nat. Rev. Genet. </span><span class="c2">2009</span><span class="c0">, </span><span class="c1">10</span><span class="c0">, 57. [27] A.-E. Saliba, A. J. Westermann, S. A. Gorski, J. Vogel, </span><span class="c1">Nucleic Acids Res. </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">42</span><span class="c0">, 8845. </span></p><p class="c181"><span class="c0">[28] J. D. Buenrostro, B. Wu, H. Y. Chang, W. J. Greenleaf, </span><span class="c1">Curr. Protoc. Mol. Biol. </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">109</span><span class="c0">, 21.29.1. </span></p><p class="c73 c252"><span class="c0">[29] P. J. Park, </span><span class="c1">Nat. Rev. Genet. </span><span class="c2">2009</span><span class="c0">, </span><span class="c1">10</span><span class="c0">, 669. </span></p><p class="c28 c252"><span class="c0">[30] D. J. Burgess, </span><span class="c1">Nat. Rev. Genet. </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">20</span><span class="c0">, 317. </span></p><p class="c103"><span class="c0">[31] A. Conesa, P. Madrigal, S. Tarazona, D. Gomez-Cabrero, A. Cervera, A. McPherson, M. W. Szcze&acute;sniak, D. J. Gaffney, L. L. Elo, X. Zhang, </span><span class="c1">Genome Biol. </span><span class="c2">2016</span><span class="c0">, </span><span class="c1">17</span><span class="c0">, 1. </span></p><p class="c234"><span class="c0">[32] L. Wang, S. Wang, W. Li, </span><span class="c1">Bioinformatics </span><span class="c2">2012</span><span class="c0">, </span><span class="c1">28</span><span class="c0">, 2184. [33] M. Martin, </span><span class="c1">EMBnet. journal </span><span class="c2">2011</span><span class="c0">, </span><span class="c1">17</span><span class="c0">, 10. </span></p><p class="c114"><span class="c0">[34] A. Dobin, T. R. Gingeras, </span><span class="c1">Curr. Protoc. Bioinform. </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">51</span><span class="c0">, 11.14.1. [35] C. Trapnell, L. Pachter, S. L. Salzberg, </span><span class="c1">Bioinformatics </span><span class="c2">2009</span><span class="c0">, </span><span class="c1">25</span><span class="c0">, 1105. [36] Y. Liao, G. K. Smyth, W. Shi, </span><span class="c1">Bioinformatics </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">30</span><span class="c0">, 923. [37] F. Rapaport, R. Khanin, Y. Liang, M. Pirun, A. Krek, P. Zumbo, C. E. </span></p><p class="c172"><span class="c0">Mason, N. D. Socci, D. Betel, </span><span class="c1">Genome Biol. </span><span class="c2">2013</span><span class="c0">, </span><span class="c1">14</span><span class="c0">, R95. [38] S. Shen, J. W. Park, Z.-x. Lu, L. Lin, M. D. Henry, Y. N. Wu, Q. Zhou, Y. Xing, </span><span class="c1">Proc. Natl. Acad. Sci. USA </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">111</span><span class="c0">, E5593. </span></p><p class="c187"><span class="c0">[39] Y. Katz, E. T. Wang, E. M. Airoldi, C. B. Burge, </span><span class="c1">Nat. Methods </span><span class="c2">2010</span><span class="c0">, </span><span class="c1">7</span><span class="c0">, 1009. </span></p><p class="c146"><span class="c0">[40] X. Wang, M. J. Cairns, presented at </span><span class="c1">BMC bioinformatics</span><span class="c0">, </span><span class="c2">2013</span><span class="c0">. [41] R. Thomas, S. Thomas, A. K. Holloway, K. S. Pollard, </span><span class="c1">Briefings in bioinformatics </span><span class="c2">2017</span><span class="c0">, </span><span class="c1">18</span><span class="c0">, 441. </span></p><p class="c24"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (13 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c70"><img><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c28 c134"><span class="c0">[42] T. L. Bailey, </span><span class="c1">Bioinformatics </span><span class="c2">2011</span><span class="c0">, </span><span class="c1">27</span><span class="c0">, 1653. </span></p><p class="c217"><span class="c0">[43] G. Yu, L.-G. Wang, Q.-Y. He, </span><span class="c1">Bioinformatics </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">31</span><span class="c0">, 2382. [44] D. Reska, M. Czajkowski, K. Jurczuk, C. Boldak, W. Kwedlo, W. Bauer, J. Koszelew, M. Kretowski, </span><span class="c1">biocybernetics and biomedical engineering </span><span class="c2">2021</span><span class="c0">, </span><span class="c1">41</span><span class="c0">, 1646. </span></p><p class="c45"><span class="c0">[45] S. X. Ge, E. W. Son, R. Yao, </span><span class="c1">BMC Bioinformatics </span><span class="c2">2018</span><span class="c0">, </span><span class="c1">19</span><span class="c0">, 534. [46] A. Jiang, K. Lehnert, L. You, R. G. Snell, </span><span class="c1">Nucleic Acids Res. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">50</span><span class="c0">, W427. </span></p><p class="c42"><span class="c0">[47] X. Li, C. Xiao, J. Qi, W. Xue, X. Xu, Z. Mu, J. Zhang, C.-Y. Li, W. Ding, </span><span class="c1">Nucleic Acids Res. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">51</span><span class="c0">, W560. </span></p><p class="c124"><span class="c0">[48] J. Zhou, S. Chen, Y. Wu, H. Li, B. Zhang, L. Zhou, Y. Hu, Z. Xiang, Z. Li, N. Chen, </span><span class="c1">Sci. Adv. </span><span class="c2">2024</span><span class="c0">, </span><span class="c1">10</span><span class="c0">, eadh8601. </span></p><p class="c215"><span class="c0">[49] S. Roy, C. Coldren, A. Karunamurthy, N. S. Kip, E. W. Klee, S. E. Lincoln, A. Leon, M. Pullambhatla, R. L. Temple-Smolkin, K. V. Voelkerding, </span><span class="c1">The Journal of Molecular Diagnostics </span><span class="c2">2018</span><span class="c0">, </span><span class="c1">20</span><span class="c0">, 4. </span></p><p class="c230"><span class="c0">[50] P. A. Ewels, A. Peltzer, S. Fillinger, H. Patel, J. Alneberg, A. Wilm, M. U. Garcia, P. Di Tommaso, S. Nahnsen, </span><span class="c1">Nat. Biotechnol. </span><span class="c2">2020</span><span class="c0">, </span><span class="c1">38</span><span class="c0">, 276. </span></p><p class="c134 c136"><span class="c0">[51] L. Wratten, A. Wilm, J. G&ouml;ke, </span><span class="c1">Nat. Methods </span><span class="c2">2021</span><span class="c0">, </span><span class="c1">18</span><span class="c0">, 1161. [52] E. B. I&cedil;s&#305;k, M. D. Brazas, R. Schwartz, B. Gaeta, P. M. Palagi, C. W. van Gelder, P. Suravajhala, H. Singh, S. L. Morgan, H. Zahroh, </span><span class="c1">Nat. Biotechnol. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">41</span><span class="c0">, 1171. </span></p><p class="c200"><span class="c0">[53] T. K. Attwood, S. Blackford, M. D. Brazas, A. Davies, M. V. Schneider, </span><span class="c1">Briefings in Bioinformatics </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">20</span><span class="c0">, 398. </span></p><p class="c87"><span class="c0">[54] J. Wei, Y. Tay, R. Bommasani, C. Raffel, B. Zoph, S. Borgeaud, D. Yogatama, M. Bosma, D. Zhou, D. Metzler, </span><span class="c1">arXiv </span><span class="c2">2022</span><span class="c0">, arXiv:2206.07682. </span></p><p class="c98"><span class="c0">[55] A. J. Thirunavukarasu, D. S. J. Ting, K. Elangovan, L. Gutierrez, T. F. Tan, D. S. W. Ting, </span><span class="c1">Nat. Med. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">29</span><span class="c0">, 1930. </span></p><p class="c256"><span class="c0">[56] A. Madani, B. Krause, E. R. Greene, S. Subramanian, B. P. Mohr, J. M. Holton, J. L. Olmos, C. Xiong, Z. Z. Sun, R. Socher, </span><span class="c1">Nat. Biotechnol. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">41</span><span class="c0">, 1099. </span></p><p class="c86"><span class="c0">[57] B. Mesk&oacute;, E. J. Topol, </span><span class="c1">npj digital medicine </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">6</span><span class="c0">, 120. [58] S. Wang, Z. Zhao, X. Ouyang, Q. Wang, D. Shen, </span><span class="c1">arXiv </span><span class="c2">2023</span><span class="c0">, arXiv:2302.07257. </span></p><p class="c67"><span class="c0">[59] J. Zhou, X. He, L. Sun, J. Xu, X. Chen, Y. Chu, L. Zhou, X. Liao, B. Zhang, X. Gao, Pre-trained multimodal large language model en hances dermatological diagnosis using SkinGPT-4. Nature Commu nications </span><span class="c1">arXiv </span><span class="c2">2024</span><span class="c0">, </span><span class="c1">15</span><span class="c0">, 5649. </span></p><p class="c233 c134 c296"><span class="c0">[60] J. Zhou, X. Chen, X. Gao, </span><span class="c1">medRxiv </span><span class="c2">2023</span><span class="c0">, 2023.06. 23.23291802. [61] T. Tu, S. Azizi, D. Driess, M. Schaekermann, M. Amin, P.-C. Chang, A. Carroll, C. Lau, R. Tanno, I. Ktena, </span><span class="c1">NEJM AI </span><span class="c2">2024</span><span class="c0">, </span><span class="c1">1</span><span class="c0">, AIoa2300138. [62] D. Flam-Shepherd, K. Zhu, A. Aspuru-Guzik, </span><span class="c1">Nat. Commun. </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">13</span><span class="c0">, 3293. </span></p><p class="c202"><span class="c0">[63] E. Shue, L. Liu, B. Li, Z. Feng, X. Li, G. Hu, </span><span class="c1">Quantitative Biology </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">11</span><span class="c0">, 105. </span></p><p class="c137"><span class="c0">[64] S. R. Piccolo, P. Denny, A. Luxton-Reilly, S. Payne, P. G. Ridge, </span><span class="c1">arXiv </span><span class="c2">2023</span><span class="c0">, arXiv:2303.13528. </span></p><p class="c39"><span class="c0">[65] L. Giray, </span><span class="c1">Ann. Biomed. Eng. </span><span class="c2">2023</span><span class="c0">, </span><span class="c1">51</span><span class="c0">, 2629. </span></p><p class="c59"><span class="c0">[66] S. Gravitas, Auto-GPT: An autonomous GPT-4 experiment </span><span class="c2">2023</span><span class="c0">. [67] B. Roziere, J. Gehring, F. Gloeckle, S. Sootla, I. Gat, X. E. Tan, Y. Adi, J. Liu, T. Remez, J. Rapin, </span><span class="c1">arXiv </span><span class="c2">2023</span><span class="c0">, arXiv:2308.12950. [68] H. Touvron, L. Martin, K. Stone, P. Albert, A. Almahairi, Y. Babaei, N. Bashlykov, S. Batra, P. Bhargava, S. Bhosale, </span><span class="c1">arXiv </span><span class="c2">2023</span><span class="c0">, arXiv:2307.09288. </span></p><p class="c11"><span class="c0">[69] C. Benner, T. van der Meulen, E. Cac&eacute;res, K. Tigyi, C. J. Donaldson, M. O. Huising, </span><span class="c1">BMC Genomics </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">15</span><span class="c0">, 620. </span></p><p class="c134 c227"><span class="c0">[70] A. Dobin, C. A. Davis, F. Schlesinger, J. Drenkow, C. Zaleski, S. Jha, P. Batut, M. Chaisson, T. R. Gingeras, </span><span class="c1">Bioinformatics </span><span class="c2">2013</span><span class="c0">, </span><span class="c1">29</span><span class="c0">, 15. [71] D. Kim, J. M. Paggi, C. Park, C. Bennett, S. L. Salzberg, </span><span class="c1">Nat. Biotech nol. </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">37</span><span class="c0">, 907. </span></p><p class="c39"><span class="c0">[72] H. Li, </span><span class="c1">Bioinformatics </span><span class="c2">2018</span><span class="c0">, </span><span class="c1">34</span><span class="c0">, 3094. </span></p><p class="c65"><span class="c0">[73] X. Chen, O. Schulz-Trieglaff, R. Shaw, B. Barnes, F. Schlesinger, M. K&auml;llberg, A. J. Cox, S. Kruglyak, C. T. Saunders, </span><span class="c1">Bioinformatics </span><span class="c2">2016</span><span class="c0">, </span><span class="c1">32</span><span class="c0">, 1220. </span></p><p class="c282"><span class="c0">[74] K. Ye, M. H. Schulz, Q. Long, R. Apweiler, Z. Ning, </span><span class="c1">Bioinformatics </span><span class="c2">2009</span><span class="c0">, </span><span class="c1">25</span><span class="c0">, 2865. </span></p><p class="c143"><span class="c0">[75] Z. H. Kwok, B. Zhang, X. H. Chew, J. J. Chan, V. Teh, H. Yang, D. Kappei, Y. Tay, </span><span class="c1">Cancer Res. </span><span class="c2">2021</span><span class="c0">, </span><span class="c1">81</span><span class="c0">, 1308. </span></p><p class="c73 c147"><span class="c0">[76] A. Bolger, F. Giorgi, </span><span class="c1">Bioinformatics </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">30</span><span class="c0">, 2114. </span></p><p class="c284"><span class="c0">[77] A. Bankevich, S. Nurk, D. Antipov, A. A. Gurevich, M. Dvorkin, A. S. Kulikov, V. M. Lesin, S. I. Nikolenko, S. Pham, A. D. Prjibelski, </span><span class="c1">J. Comput. Biol. </span><span class="c2">2012</span><span class="c0">, </span><span class="c1">19</span><span class="c0">, 455. </span></p><p class="c286"><span class="c0">[78] A. Gurevich, V. Saveliev, N. Vyahhi, G. Tesler, </span><span class="c1">Bioinformatics </span><span class="c2">2013</span><span class="c0">, </span><span class="c1">29</span><span class="c0">, 1072. </span></p><p class="c73 c147"><span class="c0">[79] H. Li, </span><span class="c1">arXiv </span><span class="c2">2013</span><span class="c0">, arXiv:1303.3997. </span></p><p class="c111"><span class="c0">[80] H. Li, B. Handsaker, A. Wysoker, T. Fennell, J. Ruan, N. Homer, G. Marth, G. Abecasis, R. Durbin, G. P. D. P. Subgroup, </span><span class="c1">Bioinformatics </span><span class="c2">2009</span><span class="c0">, </span><span class="c1">25</span><span class="c0">, 2078. </span></p><p class="c58"><span class="c0">[81] A. McKenna, M. Hanna, E. Banks, A. Sivachenko, K. Cibulskis, A. Kernytsky, K. Garimella, D. Altshuler, S. Gabriel, M. Daly, </span><span class="c1">Genome Res. </span><span class="c2">2010</span><span class="c0">, </span><span class="c1">20</span><span class="c0">, 1297. </span></p><p class="c147 c257"><span class="c0">[82] W. McLaren, L. Gil, S. E. Hunt, H. S. Riat, G. R. Ritchie, A. Thormann, P. Flicek, F. Cunningham, </span><span class="c1">Genome Biol. </span><span class="c2">2016</span><span class="c0">, </span><span class="c1">17</span><span class="c0">, 122. </span></p><p class="c94"><span class="c0">[83] P. Cingolani, A. Platts, L. L. Wang, M. Coon, T. Nguyen, L. Wang, S. J. Land, X. Lu, D. M. Ruden, </span><span class="c1">fly </span><span class="c2">2012</span><span class="c0">, </span><span class="c1">6</span><span class="c0">, 80. </span></p><p class="c147 c204"><span class="c0">[84] B. Langmead, S. L. Salzberg, </span><span class="c1">Nat. Methods </span><span class="c2">2012</span><span class="c0">, </span><span class="c1">9</span><span class="c0">, 357. [85] Y. Zhang, T. Liu, C. A. Meyer, J. Eeckhoute, D. S. Johnson, B. E. Bernstein, C. Nusbaum, R. M. Myers, M. Brown, W. Li, </span><span class="c1">Genome Biol. </span><span class="c2">2008</span><span class="c0">, </span><span class="c1">9</span><span class="c0">, 1. </span></p><p class="c85"><span class="c0">[86] T. L. Bailey, J. Johnson, C. E. Grant, W. S. Noble, </span><span class="c1">Nucleic Acids Res. </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">43</span><span class="c0">, W39. </span></p><p class="c108"><span class="c0">[87] A. R. Quinlan, I. M. Hall, </span><span class="c1">Bioinformatics </span><span class="c2">2010</span><span class="c0">, </span><span class="c1">26</span><span class="c0">, 841. [88] M. I. Love, W. Huber, S. Anders, </span><span class="c1">Genome Biol. </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">15</span><span class="c0">, 550. [89] U. Raudvere, L. Kolberg, I. Kuzmin, T. Arak, P. Adler, H. Peterson, J. Vilo, </span><span class="c1">Nucleic Acids Res. </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">47</span><span class="c0">, W191. </span></p><p class="c69"><span class="c0">[90] R. Ihaka, R. Gentleman, </span><span class="c1">J. computational and graphical statistics </span><span class="c2">1996</span><span class="c0">, </span><span class="c1">5</span><span class="c0">, 299. </span></p><p class="c147 c236"><span class="c0">[91] F. Krueger, S. R. Andrews, </span><span class="c1">Bioinformatics </span><span class="c2">2011</span><span class="c0">, </span><span class="c1">27</span><span class="c0">, 1571. [92] H. Thorvaldsd&oacute;ttir, J. T. Robinson, J. P. Mesirov, </span><span class="c1">Briefings in bioinfor matics </span><span class="c2">2013</span><span class="c0">, </span><span class="c1">14</span><span class="c0">, 178. </span></p><p class="c88"><span class="c0">[93] C. Y. McLean, D. Bristor, M. Hiller, S. L. Clarke, B. T. Schaar, C. B. Lowe, A. M. Wenger, G. Bejerano, </span><span class="c1">Nat. Biotechnol. </span><span class="c2">2010</span><span class="c0">, </span><span class="c1">28</span><span class="c0">, 495. [94] S. Koren, B. P. Walenz, K. Berlin, J. R. Miller, N. H. Bergman, A. M. Phillippy, </span><span class="c1">Genome Res. </span><span class="c2">2017</span><span class="c0">, </span><span class="c1">27</span><span class="c0">, 722. </span></p><p class="c88 c233"><span class="c0">[95] R. Vaser, I. Sovi&acute;c, N. Nagarajan, M. &Scaron;iki&acute;c, </span><span class="c1">Genome Res. </span><span class="c2">2017</span><span class="c0">, </span><span class="c1">27</span><span class="c0">, 737. [96] M. Kolmogorov, J. Yuan, Y. Lin, P. A. Pevzner, </span><span class="c1">Nat. Biotechnol. </span><span class="c2">2019</span><span class="c0">, </span><span class="c1">37</span><span class="c0">, 540. </span></p><p class="c69"><span class="c0">[97] R. R. Wick, M. B. Schultz, J. Zobel, K. E. Holt, </span><span class="c1">Bioinformatics </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">31</span><span class="c0">, 3350. </span></p><p class="c73 c147"><span class="c0">[98] G. Benson, </span><span class="c1">Nucleic Acids Res. </span><span class="c2">1999</span><span class="c0">, </span><span class="c1">27</span><span class="c0">, 573. </span></p><p class="c288"><span class="c0">[99] C.-S. Chin, P. Peluso, F. J. Sedlazeck, M. Nattestad, G. T. Concepcion, A. Clum, C. Dunn, R. O&rsquo;Malley, R. Figueroa-Balderas, A. Morales Cruz, </span><span class="c1">Nat. Methods </span><span class="c2">2016</span><span class="c0">, </span><span class="c1">13</span><span class="c0">, 1050. </span></p><p class="c206"><span class="c0">[100] G. Mar&ccedil;ais, A. L. Delcher, A. M. Phillippy, R. Coston, S. L. Salzberg, A. Zimin, </span><span class="c1">PLoS Comput. Biol. </span><span class="c2">2018</span><span class="c0">, </span><span class="c1">14</span><span class="c0">, e1005944. </span></p><p class="c37"><span class="c0">[101] S. Anders, P. T. Pyl, W. Huber, </span><span class="c1">Bioinformatics </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">31</span><span class="c0">, 166. [102] D. Nicorici, M. &cedil;Satalan, H. Edgren, S. Kangaspeska, A. Murum&auml;gi, O. Kallioniemi, S. Virtanen, O. Kilkku, </span><span class="c1">biorxiv </span><span class="c2">2014</span><span class="c0">, 011650. [103] G. Pertea, M. Pertea, </span><span class="c1">F1000Research </span><span class="c2">2020</span><span class="c0">, 9. </span></p><p class="c14"><span class="c0">[104] Z. Xia, L. A. Donehower, T. A. Cooper, J. R. Neilson, D. A. Wheeler, E. J. Wagner, W. Li, </span><span class="c1">Nat. Commun. </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">5</span><span class="c0">, 5274. </span></p><p class="c117"><span class="c0">[105] A. C. Frazee, G. Pertea, A. E. Jaffe, B. Langmead, S. L. Salzberg, J. T. Leek, </span><span class="c1">Biorxiv </span><span class="c2">2014</span><span class="c0">, </span><span class="c1">1</span><span class="c0">, 003665. </span></p><p class="c204 c233 c290"><span class="c0">[106] Y. Gao, J. Wang, F. Zhao, </span><span class="c1">Genome Biol. </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">16</span><span class="c0">, 4. [107] J. Zhang, S. Chen, J. Yang, F. Zhao, </span><span class="c1">Nat. Commun. </span><span class="c2">2020</span><span class="c0">, </span><span class="c1">11</span><span class="c0">, 90. [108] M. D. Robinson, D. J. McCarthy, G. K. Smyth, </span><span class="c1">Bioinformatics </span><span class="c2">2010</span><span class="c0">, </span><span class="c1">26</span><span class="c0">, 139. </span></p><p class="c199"><span class="c0">[109] V. Haberle, A. R. Forrest, Y. Hayashizaki, P. Carninci, B. Lenhard, </span><span class="c1">Nucleic Acids Res. </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">43</span><span class="c0">, e51. </span></p><p class="c274"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (14 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p><p class="c70"><img><span class="c5">&nbsp;</span><span class="c75">2</span><span class="c4">1983844, 2024, 44, Downloaded from https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.202407094 by Nat Prov Indonesia, Wiley Online Library on [03/11/2025]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons Licens</span><span class="c6">e</span><span class="c18">www.advancedsciencenews.com www.advancedscience.com </span></p><p class="c163"><span class="c0">[110] S. Ghosh, C.-K. K. Chan, </span><span class="c1">Plant Bioinformatics: Methods and Protocols </span><span class="c2">2016</span><span class="c0">, </span><span class="c1">1374</span><span class="c0">, 339. </span></p><p class="c294"><span class="c0">[111] L. Calviello, N. Mukherjee, E. Wyler, H. Zauber, A. Hirsekorn, M. Selbach, M. Landthaler, B. Obermayer, U. Ohler, </span><span class="c1">Nat. Methods </span><span class="c2">2016</span><span class="c0">, </span><span class="c1">13</span><span class="c0">, 165. </span></p><p class="c190"><span class="c0">[112] R. Satija, J. A. Farrell, D. Gennert, A. F. Schier, A. Regev, </span><span class="c1">Nat. Biotech nol. </span><span class="c2">2015</span><span class="c0">, </span><span class="c1">33</span><span class="c0">, 495. </span></p><p class="c134 c182"><span class="c0">[113] F. A. Wolf, P. Angerer, F. J. Theis, </span><span class="c1">Genome Biol. </span><span class="c2">2018</span><span class="c0">, </span><span class="c1">19</span><span class="c0">, 15. [114] G. Palla, H. Spitzer, M. Klein, D. Fischer, A. C. Schaar, L. B. Kuemmerle, S. Rybakov, I. L. Ibarra, O. Holmberg, I. Virshup, </span><span class="c1">Nat. Methods </span><span class="c2">2022</span><span class="c0">, </span><span class="c1">19</span><span class="c0">, 171. </span></p><p class="c216"><span class="c0">[115] T. Biancalani, G. Scalia, L. Buffoni, R. Avasthi, Z. Lu, A. Sanger, N. Tokcan, C. R. Vanderburg, &Aring;. Segerstolpe, M. Zhang, </span><span class="c1">Nat. Methods </span><span class="c2">2021</span><span class="c0">, </span><span class="c1">18</span><span class="c0">, 1352. </span></p><p class="c220"><span class="c0">[116] D. Kessner, M. Chambers, R. Burke, D. Agus, P. Mallick, </span><span class="c1">Bioinfor matics </span><span class="c2">2008</span><span class="c0">, </span><span class="c1">24</span><span class="c0">, 2534. </span></p><p class="c170"><span class="c0">[117] M. Sturm, A. Bertsch, C. Gr&ouml;pl, A. Hildebrandt, R. Hussong, E. Lange, N. Pfeifer, O. Schulz-Trieglaff, A. Zerck, K. Reinert, </span><span class="c1">BMC Bioinformatics </span><span class="c2">2008</span><span class="c0">, </span><span class="c1">9</span><span class="c0">, 163. </span></p><p class="c214"><span class="c0">[118] T. Bald, J. Barth, A. Niehues, M. Specht, M. Hippler, C. Fufezan, </span><span class="c1">Bioinformatics </span><span class="c2">2012</span><span class="c0">, </span><span class="c1">28</span><span class="c0">, 1052. </span></p><p class="c83"><span class="c17">Adv. Sci. </span><span class="c51">2024</span><span class="c5">, </span><span class="c17">11</span><span class="c5">, 2407094 </span><span class="c30">2407094 (15 of 15) </span><span class="c5">&copy; 2024 The Author(s). Advanced Science published by Wiley-VCH GmbH </span></p></body></html>